<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8" />
    <title>ChIP_and_ATAC-seq_workshop-2019</title>
    <style>.emoji {
  max-width: 1em !important;
}
del {
  text-decoration: none;
  position: relative;
}
del::after {
  border-bottom: 1px solid black;
  content: '';
  left: 0;
  position: absolute;
  right: 0;
  top: 50%;
}
ul.contains-task-list li.task-list-item {
  position: relative;
  list-style-type: none;
}
ul.contains-task-list li.task-list-item input.task-list-item-checkbox {
  position: absolute;
  transform: translateX(-100%);
  width: 26px;
}
span.critic.comment {
  position: relative;
}
span.critic.comment::before {
  content: '\1f4ac';
  position: initial;
}
span.critic.comment > span {
  display: none;
}
span.critic.comment:hover > span {
  display: initial;
  position: absolute;
  top: 100%;
  left: 0;
  border: 1px solid;
  border-radius: 5px;
  max-height: 4em;
  overflow: auto;
}
span.critic.comment:focus > span {
  display: initial;
  text-decoration: underline;
  position: initial;
  top: auto;
  left: auto;
  border: initial;
  border-radius: initial;
}
table {
  border-collapse: collapse;
  border-spacing: 0;
  background-color: transparent;
}

body {
  overflow: initial !important;
  overflow: hidden;
  font-family: "Helvetica Neue", Helvetica, "Segoe UI", Arial, freesans, sans-serif;
  line-height: 1.6;
  word-wrap: break-word;
  padding: 30px;
  font-size: 16px;
  color: #333;
  background-color: #fff;
}
body > *:first-child {
  margin-top: 0 !important;
}
body > *:last-child {
  margin-bottom: 0 !important;
}
body a:not([href]) {
  color: inherit;
  text-decoration: none;
}
body .absent {
  color: #c00;
}
body .anchor {
  position: absolute;
  top: 0;
  left: 0;
  display: block;
  padding-right: 6px;
  padding-left: 30px;
  margin-left: -30px;
}
body .anchor:focus {
  outline: none;
}
body h1,
body h2,
body h3,
body h4,
body h5,
body h6 {
  position: relative;
  margin-top: 1em;
  margin-bottom: 16px;
  font-weight: bold;
  line-height: 1.4;
}
body h1 .octicon-link,
body h2 .octicon-link,
body h3 .octicon-link,
body h4 .octicon-link,
body h5 .octicon-link,
body h6 .octicon-link {
  display: none;
  color: #000;
  vertical-align: middle;
}
body h1:hover .anchor,
body h2:hover .anchor,
body h3:hover .anchor,
body h4:hover .anchor,
body h5:hover .anchor,
body h6:hover .anchor {
  padding-left: 8px;
  margin-left: -30px;
  text-decoration: none;
}
body h1:hover .anchor .octicon-link,
body h2:hover .anchor .octicon-link,
body h3:hover .anchor .octicon-link,
body h4:hover .anchor .octicon-link,
body h5:hover .anchor .octicon-link,
body h6:hover .anchor .octicon-link {
  display: inline-block;
}
body h1 tt,
body h2 tt,
body h3 tt,
body h4 tt,
body h5 tt,
body h6 tt,
body h1 code,
body h2 code,
body h3 code,
body h4 code,
body h5 code,
body h6 code {
  font-size: inherit;
}
body h1 {
  padding-bottom: 0.3em;
  font-size: 2.25em;
  line-height: 1.2;
  border-bottom: 1px solid #eee;
}
body h1 .anchor {
  line-height: 1;
}
body h2 {
  padding-bottom: 0.3em;
  font-size: 1.75em;
  line-height: 1.225;
  border-bottom: 1px solid #eee;
}
body h2 .anchor {
  line-height: 1;
}
body h3 {
  font-size: 1.5em;
  line-height: 1.43;
}
body h3 .anchor {
  line-height: 1.2;
}
body h4 {
  font-size: 1.25em;
}
body h4 .anchor {
  line-height: 1.2;
}
body h5 {
  font-size: 1em;
}
body h5 .anchor {
  line-height: 1.1;
}
body h6 {
  font-size: 1em;
  color: #777;
}
body h6 .anchor {
  line-height: 1.1;
}
body p,
body blockquote,
body ul,
body ol,
body dl,
body table,
body pre {
  margin-top: 0;
  margin-bottom: 16px;
}
body hr {
  height: 4px;
  padding: 0;
  margin: 16px 0;
  background-color: #e7e7e7;
  border: 0 none;
}
body ul,
body ol {
  padding-left: 2em;
}
body ul.no-list,
body ol.no-list {
  padding: 0;
  list-style-type: none;
}
body ul ul,
body ul ol,
body ol ol,
body ol ul {
  margin-top: 0;
  margin-bottom: 0;
}
body li > p {
  margin-top: 16px;
}
body dl {
  padding: 0;
}
body dl dt {
  padding: 0;
  margin-top: 16px;
  font-size: 1em;
  font-style: italic;
  font-weight: bold;
}
body dl dd {
  padding: 0 16px;
  margin-bottom: 16px;
}
body blockquote {
  padding: 0 15px;
  color: #777;
  border-left: 4px solid #ddd;
}
body blockquote > :first-child {
  margin-top: 0;
}
body blockquote > :last-child {
  margin-bottom: 0;
}
body table {
  display: block;
  width: 100%;
  overflow: auto;
  word-break: normal;
  word-break: keep-all;
}
body table th {
  font-weight: bold;
}
body table th,
body table td {
  padding: 6px 13px;
  border: 1px solid #ddd;
}
body table tr {
  background-color: #fff;
  border-top: 1px solid #ccc;
}
body table tr:nth-child(2n) {
  background-color: #f8f8f8;
}
body img {
  max-width: 100%;
  -moz-box-sizing: border-box;
  box-sizing: border-box;
}
body .emoji {
  max-width: none;
}
body span.frame {
  display: block;
  overflow: hidden;
}
body span.frame > span {
  display: block;
  float: left;
  width: auto;
  padding: 7px;
  margin: 13px 0 0;
  overflow: hidden;
  border: 1px solid #ddd;
}
body span.frame span img {
  display: block;
  float: left;
}
body span.frame span span {
  display: block;
  padding: 5px 0 0;
  clear: both;
  color: #333;
}
body span.align-center {
  display: block;
  overflow: hidden;
  clear: both;
}
body span.align-center > span {
  display: block;
  margin: 13px auto 0;
  overflow: hidden;
  text-align: center;
}
body span.align-center span img {
  margin: 0 auto;
  text-align: center;
}
body span.align-right {
  display: block;
  overflow: hidden;
  clear: both;
}
body span.align-right > span {
  display: block;
  margin: 13px 0 0;
  overflow: hidden;
  text-align: right;
}
body span.align-right span img {
  margin: 0;
  text-align: right;
}
body span.float-left {
  display: block;
  float: left;
  margin-right: 13px;
  overflow: hidden;
}
body span.float-left span {
  margin: 13px 0 0;
}
body span.float-right {
  display: block;
  float: right;
  margin-left: 13px;
  overflow: hidden;
}
body span.float-right > span {
  display: block;
  margin: 13px auto 0;
  overflow: hidden;
  text-align: right;
}
body code,
body tt {
  padding: 0;
  padding-top: 0.2em;
  padding-bottom: 0.2em;
  margin: 0;
  font-size: 85%;
  background-color: rgba(0, 0, 0, 0.04);
  border-radius: 3px;
}
body code:before,
body tt:before,
body code:after,
body tt:after {
  letter-spacing: -0.2em;
  content: "\00a0";
}
body code br,
body tt br {
  display: none;
}
body del code {
  text-decoration: inherit;
}
body pre > code {
  padding: 0;
  margin: 0;
  font-size: 100%;
  word-break: normal;
  white-space: pre;
  background: transparent;
  border: 0;
}
body .highlight {
  margin-bottom: 16px;
}
body .highlight pre,
body pre {
  padding: 16px;
  overflow: auto;
  font-size: 85%;
  line-height: 1.45;
  background-color: #f7f7f7;
  border-radius: 3px;
}
body .highlight pre {
  margin-bottom: 0;
  word-break: normal;
}
body pre {
  word-wrap: normal;
}
body pre code,
body pre tt {
  display: inline;
  max-width: initial;
  padding: 0;
  margin: 0;
  overflow: initial;
  line-height: inherit;
  word-wrap: normal;
  background-color: transparent;
  border: 0;
}
body pre code:before,
body pre tt:before,
body pre code:after,
body pre tt:after {
  content: normal;
}
body kbd {
  display: inline-block;
  padding: 3px 5px;
  font-size: 11px;
  line-height: 10px;
  color: #555;
  vertical-align: middle;
  background-color: #fcfcfc;
  border: solid 1px #ccc;
  border-bottom-color: #bbb;
  border-radius: 3px;
  box-shadow: inset 0 -1px 0 #bbb;
}
span.critic.comment > span {
  background-color: #fff;
}
a {
  color: #337ab7;
}

.bracket-matcher .region {
  border-bottom: 1px dotted lime;
  position: absolute;
}
.line-number.bracket-matcher.bracket-matcher {
  color: #c5c8c6;
  background-color: rgba(255, 255, 255, 0.14);
}

.spell-check-misspelling .region {
  border-bottom: 2px dotted rgba(255, 51, 51, 0.75);
}
.spell-check-corrections {
  width: 25em !important;
}

pre.editor-colors {
  background-color: #1d1f21;
  color: #c5c8c6;
}
pre.editor-colors .invisible-character {
  color: rgba(197, 200, 198, 0.2);
}
pre.editor-colors .indent-guide {
  color: rgba(197, 200, 198, 0.2);
}
pre.editor-colors .wrap-guide {
  background-color: rgba(197, 200, 198, 0.1);
}
pre.editor-colors .gutter {
  background-color: #292c2f;
}
pre.editor-colors .gutter .cursor-line {
  background-color: rgba(255, 255, 255, 0.14);
}
pre.editor-colors .line-number.cursor-line-no-selection {
  background-color: rgba(255, 255, 255, 0.14);
}
pre.editor-colors .gutter .line-number.folded,
pre.editor-colors .gutter .line-number:after,
pre.editor-colors .fold-marker:after {
  color: #fba0e3;
}
pre.editor-colors .invisible {
  color: #c5c8c6;
}
pre.editor-colors .cursor {
  border-color: white;
}
pre.editor-colors .selection .region {
  background-color: #444;
}
pre.editor-colors .bracket-matcher .region {
  border-bottom: 1px solid #f8de7e;
  margin-top: -1px;
  opacity: .7;
}
.syntax--comment {
  color: #8a8a8a;
}
.syntax--entity {
  color: #FFD2A7;
}
.syntax--entity.syntax--name.syntax--type {
  text-decoration: underline;
  color: #FFFFB6;
}
.syntax--entity.syntax--other.syntax--inherited-class {
  color: #9B5C2E;
}
.syntax--keyword {
  color: #96CBFE;
}
.syntax--keyword.syntax--control {
  color: #96CBFE;
}
.syntax--keyword.syntax--operator {
  color: #EDEDED;
}
.syntax--storage {
  color: #CFCB90;
}
.syntax--storage.syntax--modifier {
  color: #96CBFE;
}
.syntax--constant {
  color: #99CC99;
}
.syntax--constant.syntax--numeric {
  color: #FF73FD;
}
.syntax--variable {
  color: #C6C5FE;
}
.syntax--invalid.syntax--deprecated {
  text-decoration: underline;
  color: #FD5FF1;
}
.syntax--invalid.syntax--illegal {
  color: #FD5FF1;
  background-color: rgba(86, 45, 86, 0.75);
}
.syntax--string .syntax--source,
.syntax--string .syntax--meta.syntax--embedded.syntax--line {
  color: #EDEDED;
}
.syntax--string .syntax--punctuation.syntax--section.syntax--embedded {
  color: #00A0A0;
}
.syntax--string .syntax--punctuation.syntax--section.syntax--embedded .syntax--source {
  color: #00A0A0;
}
.syntax--string {
  color: #A8FF60;
}
.syntax--string .syntax--constant {
  color: #00A0A0;
}
.syntax--string.syntax--regexp {
  color: #E9C062;
}
.syntax--string.syntax--regexp .syntax--constant.syntax--character.syntax--escape,
.syntax--string.syntax--regexp .syntax--source.syntax--ruby.syntax--embedded,
.syntax--string.syntax--regexp .syntax--string.syntax--regexp.syntax--arbitrary-repetition {
  color: #FF8000;
}
.syntax--string.syntax--regexp.syntax--group {
  color: #C6A24F;
  background-color: rgba(255, 255, 255, 0.06);
}
.syntax--string.syntax--regexp.syntax--character-class {
  color: #B18A3D;
}
.syntax--string .syntax--variable {
  color: #8A9A95;
}
.syntax--support {
  color: #FFFFB6;
}
.syntax--support.syntax--function {
  color: #DAD085;
}
.syntax--support.syntax--constant {
  color: #FFD2A7;
}
.syntax--support.syntax--type.syntax--property-name.syntax--css {
  color: #EDEDED;
}
.syntax--source .syntax--entity.syntax--name.syntax--tag,
.syntax--source .syntax--punctuation.syntax--tag {
  color: #96CBFE;
}
.syntax--source .syntax--entity.syntax--other.syntax--attribute-name {
  color: #FF73FD;
}
.syntax--entity.syntax--other.syntax--attribute-name {
  color: #FF73FD;
}
.syntax--entity.syntax--name.syntax--tag.syntax--namespace,
.syntax--entity.syntax--other.syntax--attribute-name.syntax--namespace {
  color: #E18964;
}
.syntax--meta.syntax--preprocessor.syntax--c {
  color: #8996A8;
}
.syntax--meta.syntax--preprocessor.syntax--c .syntax--keyword {
  color: #AFC4DB;
}
.syntax--meta.syntax--cast {
  color: #676767;
}
.syntax--meta.syntax--sgml.syntax--html .syntax--meta.syntax--doctype,
.syntax--meta.syntax--sgml.syntax--html .syntax--meta.syntax--doctype .syntax--entity,
.syntax--meta.syntax--sgml.syntax--html .syntax--meta.syntax--doctype .syntax--string,
.syntax--meta.syntax--xml-processing,
.syntax--meta.syntax--xml-processing .syntax--entity,
.syntax--meta.syntax--xml-processing .syntax--string {
  color: #8a8a8a;
}
.syntax--meta.syntax--tag .syntax--entity,
.syntax--meta.syntax--tag > .syntax--punctuation,
.syntax--meta.syntax--tag.syntax--inline .syntax--entity {
  color: #FF73FD;
}
.syntax--meta.syntax--tag .syntax--name,
.syntax--meta.syntax--tag.syntax--inline .syntax--name,
.syntax--meta.syntax--tag > .syntax--punctuation {
  color: #96CBFE;
}
.syntax--meta.syntax--selector.syntax--css .syntax--entity.syntax--name.syntax--tag {
  text-decoration: underline;
  color: #96CBFE;
}
.syntax--meta.syntax--selector.syntax--css .syntax--entity.syntax--other.syntax--attribute-name.syntax--tag.syntax--pseudo-class {
  color: #8F9D6A;
}
.syntax--meta.syntax--selector.syntax--css .syntax--entity.syntax--other.syntax--attribute-name.syntax--id {
  color: #8B98AB;
}
.syntax--meta.syntax--selector.syntax--css .syntax--entity.syntax--other.syntax--attribute-name.syntax--class {
  color: #62B1FE;
}
.syntax--meta.syntax--property-group .syntax--support.syntax--constant.syntax--property-value.syntax--css,
.syntax--meta.syntax--property-value .syntax--support.syntax--constant.syntax--property-value.syntax--css {
  color: #F9EE98;
}
.syntax--meta.syntax--preprocessor.syntax--at-rule .syntax--keyword.syntax--control.syntax--at-rule {
  color: #8693A5;
}
.syntax--meta.syntax--property-value .syntax--support.syntax--constant.syntax--named-color.syntax--css,
.syntax--meta.syntax--property-value .syntax--constant {
  color: #87C38A;
}
.syntax--meta.syntax--constructor.syntax--argument.syntax--css {
  color: #8F9D6A;
}
.syntax--meta.syntax--diff,
.syntax--meta.syntax--diff.syntax--header {
  color: #F8F8F8;
  background-color: #0E2231;
}
.syntax--meta.syntax--separator {
  color: #60A633;
  background-color: #242424;
}
.syntax--meta.syntax--line.syntax--entry.syntax--logfile,
.syntax--meta.syntax--line.syntax--exit.syntax--logfile {
  background-color: rgba(238, 238, 238, 0.16);
}
.syntax--meta.syntax--line.syntax--error.syntax--logfile {
  background-color: #751012;
}
.syntax--source.syntax--gfm {
  color: #999;
}
.syntax--gfm .syntax--markup.syntax--heading {
  color: #eee;
}
.syntax--gfm .syntax--link {
  color: #555;
}
.syntax--gfm .syntax--variable.syntax--list,
.syntax--gfm .syntax--support.syntax--quote {
  color: #555;
}
.syntax--gfm .syntax--link .syntax--entity {
  color: #ddd;
}
.syntax--gfm .syntax--raw {
  color: #aaa;
}
.syntax--markdown .syntax--paragraph {
  color: #999;
}
.syntax--markdown .syntax--heading {
  color: #eee;
}
.syntax--markdown .syntax--raw {
  color: #aaa;
}
.syntax--markdown .syntax--link {
  color: #555;
}
.syntax--markdown .syntax--link .syntax--string {
  color: #555;
}
.syntax--markdown .syntax--link .syntax--string.syntax--title {
  color: #ddd;
}

/*
 * Your Stylesheet
 *
 * This stylesheet is loaded when Atom starts up and is reloaded automatically
 * when it is changed and saved.
 *
 * Add your own CSS or Less to fully customize Atom.
 * If you are unfamiliar with Less, you can read more about it here:
 * http://lesscss.org
 */
/*
 * Examples
 * (To see them, uncomment and save)
 */
</style>

  </head>
  <body>
    <h1>ChIP and ATAC-seq workshop 2019</h1>
<h3>-by Jonas “your-friendly-neighbourhood-chromatin-companion” Ungerbäck</h3>
<p>##########################################################################</p>
<h3>Workshop information:</h3>
<p>In this workshop you will learn to work with ChIP and ATAC-sequencing data, mainly via the command line interface. You will learn to call peaks, looking at overlapping peaks between different datasets, annotate peaks, motif analysis with Homer and the MEME suite and creating bigwigs and heatmaps with deepTools.</p>
<hr>
<h5>These are the tools you primarily will use for the workshop:</h5>
<p>HOMER: <a href="http://homer.ucsd.edu/homer/">http://homer.ucsd.edu/homer/</a></p>
<p>BedTools: <a href="https://bedtools.readthedocs.io/en/latest/">https://bedtools.readthedocs.io/en/latest/</a></p>
<p>MACS2: <a href="https://github.com/taoliu/MACS">https://github.com/taoliu/MACS</a></p>
<p>MEME: <a href="http://meme-suite.org/">http://meme-suite.org/</a></p>
<p>DeepTools: <a href="https://deeptools.readthedocs.io/en/develop/">https://deeptools.readthedocs.io/en/develop/</a></p>
<hr>
<hr>
<h3>The datasets</h3>
<p>You will be working with ChIP -and ATAC-datasets from T-cell development in mouse. If you are interested you can read the original papers here:</p>
<p><a href="https://www.ncbi.nlm.nih.gov/pubmed/22500808">https://www.ncbi.nlm.nih.gov/pubmed/22500808</a> (ChIP-seq datasets)
<a href="https://www.ncbi.nlm.nih.gov/pubmed/29466756">https://www.ncbi.nlm.nih.gov/pubmed/29466756</a> (ATAC-seq datasets)</p>
<h4>The datasets are as follows:</h4>
<h5>ChIP (single-end sequencing data)</h5>
<p>The transcription factor PU.1 as well as the active promoter/enhancer H3K4me2 marker.</p>
<h4>ATAC (paired-end sequencing data)</h4>
<p>ProT cells pre (ETPs) and post (DN3) T-cell lineage commitment</p>
<p>You will recieve data in the bed-format but the command for the preprocessing  can be found in the following scripts:</p>
<p><code style="font-family: Menlo, Consolas, &quot;DejaVu Sans Mono&quot;, monospace;">preprocessing_ChIP.sh</code> <code style="font-family: Menlo, Consolas, &quot;DejaVu Sans Mono&quot;, monospace;">ChIP_wrapper.sh</code> <code style="font-family: Menlo, Consolas, &quot;DejaVu Sans Mono&quot;, monospace;">preprocessing_ATAC.sh</code> <code style="font-family: Menlo, Consolas, &quot;DejaVu Sans Mono&quot;, monospace;">ATAC_wrapper.sh</code> <code style="font-family: Menlo, Consolas, &quot;DejaVu Sans Mono&quot;, monospace;">batch_maketagdirectory.sh</code> <code style="font-family: Menlo, Consolas, &quot;DejaVu Sans Mono&quot;, monospace;">makeTag_wrapper.sh</code>.</p>
<h1></h1>
<hr>
<h3>First and foremost:</h3>
<p>We will use aurora and not Lsens2 for this workshops so after login into Aurora (<a href="http://aurora.lunarc.lu.se">aurora.lunarc.lu.se</a>), go to the folder you will be working in /lunarc/nobackup/users/<em>youruser</em>.</p>
<p>Copy the following folder to you user:</p>
<pre class="editor-colors lang-bash"><span><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">cp</span> <span class="syntax--entity syntax--other syntax--attribute-name">-r</span> /lunarc/nobackup/users/jonun/ChIP-ATAC-workshop-2019-course-material .</span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#Go to the folder.</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function"><span class="syntax--support syntax--function">cd</span></span> ChIP-ATAC-workshop-2019-course-material</span></span></pre>
<p>###############################################################
###############################################################</p>
<h3>ChIP-seq assignments</h3>
<p><strong>1.</strong> We will user Homer to call ChIP-peaks but before we do that we have to create a file-format Homer can work with. This is called a Tag Directory.</p>
<p>Start by loading the required modules.</p>
<pre class="editor-colors lang-bash"><span class=""><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">module</span> load GCC/7.3.0-2.30 homer/4.10</span></span></pre>
<p>Run the following to check what options makeTagDirectory has</p>
<pre class="editor-colors lang-bash"><span class=""><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">makeTagDirectory</span></span></span></pre>
<p>We will now use Homer’s makeTagDirectory to create tag directories from all the filtered bam-files (including ATAC-samples since this will be used later). Sadly this will not immediately work since Homer requires samtools to this. Batch scripting to the rescue.</p>
<pre class="editor-colors lang-bash"><span><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#! /bin/bash</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#SBATCH -A lu2018-2-44</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#SBATCH -p lu</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#SBATCH --reservation=lu2018-2-44</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#SBATCH -n 4 # how many processor cores to use</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#SBATCH -N 1 # how many processors to use (always use 1 here unless you know what you are doing)</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#SBATCH -t 00:20:00 # kill the job after ths hh::mm::ss time</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#SBATCH -J 'makeTagDirectories' # name of the job</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#SBATCH -o 'makeTagDirectories%j.out' # stdout log file</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#SBATCH -e 'makeTagDirectories%j.err' # stderr log file</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#SBATCH -p dell # which partition to use</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block"># A script that preprocesses the ChIP samples.</span></span></span>
<span class=""></span> 
<span class=""></span> 
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#Go to sample-directory</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function"><span class="syntax--support syntax--function">cd</span></span> ~/NAS2/ChIP-ATAC-workshop-2019-course-material/<span class="syntax--punctuation syntax--section syntax--embedded">${</span><span class="syntax--variable syntax--other syntax--member">sample</span><span class="syntax--punctuation syntax--section syntax--embedded">}</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#Load the samtools module</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">module</span> load GCC/7.3.0-2.30 SAMtools/1.9</span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#First create a sam-file</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">samtools</span> view <span class="syntax--entity syntax--other syntax--attribute-name">-@</span> 4 <span class="syntax--entity syntax--other syntax--attribute-name">-h</span> <span class="syntax--entity syntax--other syntax--attribute-name">-o</span> <span class="syntax--punctuation syntax--section syntax--embedded">${</span><span class="syntax--variable syntax--other syntax--member">sample</span><span class="syntax--punctuation syntax--section syntax--embedded">}</span>.sam <span class="syntax--punctuation syntax--section syntax--embedded">${</span><span class="syntax--variable syntax--other syntax--member">sample</span><span class="syntax--punctuation syntax--section syntax--embedded">}</span>.filt.bam</span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">module</span> purge</span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#Load the Homer module:</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">module</span> load GCC/7.3.0-2.30 homer/4.10</span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#Create a tagdirectory</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">makeTagDirectory</span> <span class="syntax--punctuation syntax--section syntax--embedded">${</span><span class="syntax--variable syntax--other syntax--member">sample</span><span class="syntax--punctuation syntax--section syntax--embedded">}</span>_tagdir <span class="syntax--punctuation syntax--section syntax--embedded">${</span><span class="syntax--variable syntax--other syntax--member">sample</span><span class="syntax--punctuation syntax--section syntax--embedded">}</span>.sam</span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">rm</span> <span class="syntax--punctuation syntax--section syntax--embedded">${</span><span class="syntax--variable syntax--other syntax--member">sample</span><span class="syntax--punctuation syntax--section syntax--embedded">}</span>.sam</span></span></pre>
<p>Create a wrapper script that loop over the samples.</p>
<pre class="editor-colors lang-bash"><span><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#!/bin/bash</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#A wrapper that loops over the samples and calls batch_maketagdirectory.sh for each sample.</span></span></span>
<span class=""></span> 
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--variable syntax--other syntax--member">samples</span>=<span class="syntax--string">"10340_DN1_H3K4me2 10500_DN3_H3K4me2 11046_DN1_H3K4me2 11307_DN1_PU1 11353_DN3_H3K4me2 11735_DN2b_PU1 DN1_input ThyDN3_input ATAC-seq_DN3_T_cells_rep1 ATAC-seq_DN3_T_cells_rep2 ATAC-seq_ETP_T_cells_rep1 ATAC-seq_ETP_T_cells_rep2"</span></span></span>
<span class=""></span> 
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--keyword syntax--control">for</span> <span class="syntax--variable syntax--other syntax--member">i</span> <span class="syntax--keyword syntax--control">in</span> <span class="syntax--punctuation syntax--section syntax--embedded">${</span><span class="syntax--variable syntax--other syntax--member">samples</span><span class="syntax--punctuation syntax--section syntax--embedded">}</span>;<span class="syntax--keyword syntax--control">do</span></span></span>
<span class=""></span> 
<span class=""><span class="syntax--source syntax--shell"><span class="leading-whitespace">        </span><span class="syntax--entity syntax--name syntax--function"><span class="syntax--support syntax--function">echo</span></span> <span class="syntax--string">"sbatch --export=sample=<span class="syntax--punctuation syntax--section syntax--embedded">${</span><span class="syntax--variable syntax--other syntax--member">i</span><span class="syntax--punctuation syntax--section syntax--embedded">}</span> batch_maketagdirectory.sh"</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="leading-whitespace">        </span><span class="syntax--entity syntax--name syntax--function">sbatch</span> <span class="syntax--entity syntax--other syntax--attribute-name">--export=sample=</span><span class="syntax--punctuation syntax--section syntax--embedded">${</span><span class="syntax--variable syntax--other syntax--member">i</span><span class="syntax--punctuation syntax--section syntax--embedded">}</span> batch_maketagdirectory.sh</span></span>
<span class=""></span> 
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--keyword syntax--control">done</span></span></span></pre>
<p>Call the script</p>
<pre class="editor-colors lang-bash"><span class=""><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">bash</span> makeTag_wrapper.sh</span></span></pre>
<p>Since this takes a while I have prepared it so in each folder you will find a subfolder called a something-something tagdir <img class="emoji" draggable="false" alt="😃" src="/Users/Jonas/.atom/packages/markdown-preview-plus/node_modules/twemoji/2/svg/1f603.svg">. Alternatively, to circumvent samtools at this step, we could have created a bed-file and then made the tagdir from that.
You can read more about tag directories here:
<a href="http://homer.ucsd.edu/homer/ngs/tagDir.html">http://homer.ucsd.edu/homer/ngs/tagDir.html</a></p>
<p><strong>We are now ready to call ChIP-peaks.</strong></p>
<p><strong>2.</strong> <strong>Peak finding</strong>
Run the following to check what options findPeaks has:</p>
<pre class="editor-colors lang-bash"><span><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">module</span> load GCC/7.3.0-2.30 homer/4.10</span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">findPeaks</span></span></span></pre>
<p>As you can see findPeaks can be customized with a large variety of options but by setting <code style="font-family: Menlo, Consolas, &quot;DejaVu Sans Mono&quot;, monospace;">-style</code> to either <code style="font-family: Menlo, Consolas, &quot;DejaVu Sans Mono&quot;, monospace;">factor</code> or <code style="font-family: Menlo, Consolas, &quot;DejaVu Sans Mono&quot;, monospace;">histone</code>, presets several options that are usually fine.</p>
<pre class="editor-colors lang-bash"><span class=""><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">findPeaks</span> 10340_DN1_H3K4me2/10340_DN1_H3K4me2_tagdir/ <span class="syntax--entity syntax--other syntax--attribute-name">-style</span> histone <span class="syntax--entity syntax--other syntax--attribute-name">-o</span> auto <span class="syntax--entity syntax--other syntax--attribute-name">-i</span> DN1_input/DN1_input_tagdir/</span></span></pre>
<p>This will create a file in the tagdirectory called <code style="font-family: Menlo, Consolas, &quot;DejaVu Sans Mono&quot;, monospace;">peaks.txt</code> or <code style="font-family: Menlo, Consolas, &quot;DejaVu Sans Mono&quot;, monospace;">regions.txt</code>.</p>
<p><strong>Task: Do this for all ChIP-samples (differentiate between histones and factors). I would recommend using a sbatch script.</strong></p>
<p>How many peaks/regions did we get for each sample? <strong>Hint:</strong> Check the peak/region files in each tagdir with <code style="font-family: Menlo, Consolas, &quot;DejaVu Sans Mono&quot;, monospace;">head peak_file</code>. A good praxis is also to rename the peak-file to something descriptive.</p>
<blockquote>
<p>10340_DN1_H3K4me2_regions.txt: Total peaks = 45085
11046_DN1_H3K4me2_regions.txt: Total peaks = 42673
10500_DN3_H3K4me2_regions.txt: Total peaks = 28009
11353_DN3_H3K4me2_regions.txt: Total peaks = 32967
11307_DN1_PU1_peaks.txt:       Total peaks = 51993
11735_DN2b_PU1_peaks.txt:      Total peaks = 17777</p>
</blockquote>
<p>This is a good start and but no matter how good a peak caller is, extra filtering is often required. If replicates exist, IDR (<a href="https://github.com/nboley/idris">https://github.com/nboley/idris</a>) is the state-of-the-art method for peak reproducibility. It is however relatively cumbersome to use and outside the scope of this one-day-course. <strong>What to do instead?</strong></p>
<p><strong>Task:</strong>
<strong>If NO replicates:</strong> Filter for a Homer Normalized peak-score (column #6). Empirically I have found that when TF-ChIP a score of ≥ 10-15 is fine. The lower the score the higher the likelihood that we are picking up noise.
<strong>Problem:</strong> The hashtags (#) on the first lines of the Homer peak file. Possible to remove in Excel but faster to use the combined power of <code style="font-family: Menlo, Consolas, &quot;DejaVu Sans Mono&quot;, monospace;">sed</code> and <code style="font-family: Menlo, Consolas, &quot;DejaVu Sans Mono&quot;, monospace;">awk</code>:</p>
<p>Example:  <code style="font-family: Menlo, Consolas, &quot;DejaVu Sans Mono&quot;, monospace;">sed '/^#/d' 11735_DN2b_PU1_peaks.txt|awk 'BEGIN {FS=OFS="\t"} ; {if ($6 &gt;= 15) print }' - &gt; 11735_DN2b_PU1_peaks_15.txt</code>
$6 is the column where Homer stores the Normalized peak score in a peak file.</p>
<blockquote>
<p>11307_DN1_PU1_peaks_15.txt : 37476 peaks
11735_DN2b_PU1_peaks_15.txt: 6688 peaks</p>
</blockquote>
<p><strong>If replicates:</strong> Later we will use BedTools but for now we will use Homer’s <code style="font-family: Menlo, Consolas, &quot;DejaVu Sans Mono&quot;, monospace;">mergePeaks</code> in combination with the <code style="font-family: Menlo, Consolas, &quot;DejaVu Sans Mono&quot;, monospace;">-prefix</code> option.</p>
<p>Type:</p>
<pre class="editor-colors lang-bash"><span class=""><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">mergePeaks</span></span></span></pre>
<p>Create a new folder to merge the replicates in from the H3K4me2 samples and copy the peak files here:</p>
<pre class="editor-colors lang-bash"><span><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">mergePeaks</span> 10340_DN1_H3K4me2_regions.txt 11046_DN1_H3K4me2_regions.txt <span class="syntax--entity syntax--other syntax--attribute-name">-prefix</span> DN1</span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="hard-tab leading-whitespace">  </span><span class="syntax--entity syntax--name syntax--function">Max</span> distance to merge: direct overlap required (<span class="syntax--entity syntax--name syntax--function">-d</span> given)</span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="hard-tab leading-whitespace">  </span><span class="syntax--entity syntax--name syntax--function">Merging</span> peaks...</span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="hard-tab leading-whitespace">  </span><span class="syntax--entity syntax--name syntax--function">Comparing</span> 10340_DN1_H3K4me2_regions.txt (<span class="syntax--entity syntax--name syntax--function">45085</span> total) <span class="syntax--entity syntax--name syntax--function">and</span> 10340_DN1_H3K4me2_regions.txt (<span class="syntax--entity syntax--name syntax--function">45085</span> total)</span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="hard-tab leading-whitespace">  </span><span class="syntax--entity syntax--name syntax--function">Comparing</span> 10340_DN1_H3K4me2_regions.txt (<span class="syntax--entity syntax--name syntax--function">45085</span> total) <span class="syntax--entity syntax--name syntax--function">and</span> 11046_DN1_H3K4me2_regions.txt (<span class="syntax--entity syntax--name syntax--function">42673</span> total)</span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="hard-tab leading-whitespace">  </span><span class="syntax--entity syntax--name syntax--function">Comparing</span> 11046_DN1_H3K4me2_regions.txt (<span class="syntax--entity syntax--name syntax--function">42673</span> total) <span class="syntax--entity syntax--name syntax--function">and</span> 10340_DN1_H3K4me2_regions.txt (<span class="syntax--entity syntax--name syntax--function">45085</span> total)</span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="hard-tab leading-whitespace">  </span><span class="syntax--entity syntax--name syntax--function">Comparing</span> 11046_DN1_H3K4me2_regions.txt (<span class="syntax--entity syntax--name syntax--function">42673</span> total) <span class="syntax--entity syntax--name syntax--function">and</span> 11046_DN1_H3K4me2_regions.txt (<span class="syntax--entity syntax--name syntax--function">42673</span> total)</span></span>
<span class=""></span> 
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">10340_DN1_H3K4me2_regions.txt</span><span class="hard-tab"> </span>11046_DN1_H3K4me2_regions.txt<span class="hard-tab"> </span>Total<span class="hard-tab"> </span>Name</span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="hard-tab leading-whitespace">  </span><span class="syntax--entity syntax--name syntax--function">X</span><span class="hard-tab"> </span>4530<span class="hard-tab">  </span>11046_DN1_H3K4me2_regions.txt</span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">X</span><span class="hard-tab"> </span><span class="hard-tab">  </span>7874<span class="hard-tab">  </span>10340_DN1_H3K4me2_regions.txt</span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">X</span><span class="hard-tab"> </span>X<span class="hard-tab"> </span>36998<span class="hard-tab"> </span>10340_DN1_H3K4me2_regions.txt<span class="syntax--keyword syntax--operator">|</span><span class="syntax--entity syntax--name syntax--function">11046_DN1_H3K4me2_regions.txt</span></span></span></pre>
<pre class="editor-colors lang-bash"><span><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">mergePeaks</span> 10500_DN3_H3K4me2_regions.txt 11353_DN3_H3K4me2_regions.txt <span class="syntax--entity syntax--other syntax--attribute-name">-prefix</span> DN3</span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="hard-tab leading-whitespace">  </span><span class="syntax--entity syntax--name syntax--function">Max</span> distance to merge: direct overlap required (<span class="syntax--entity syntax--name syntax--function">-d</span> given)</span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="hard-tab leading-whitespace">  </span><span class="syntax--entity syntax--name syntax--function">Merging</span> peaks...</span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="hard-tab leading-whitespace">  </span><span class="syntax--entity syntax--name syntax--function">Comparing</span> 10500_DN3_H3K4me2_regions.txt (<span class="syntax--entity syntax--name syntax--function">28009</span> total) <span class="syntax--entity syntax--name syntax--function">and</span> 10500_DN3_H3K4me2_regions.txt (<span class="syntax--entity syntax--name syntax--function">28009</span> total)</span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="hard-tab leading-whitespace">  </span><span class="syntax--entity syntax--name syntax--function">Comparing</span> 10500_DN3_H3K4me2_regions.txt (<span class="syntax--entity syntax--name syntax--function">28009</span> total) <span class="syntax--entity syntax--name syntax--function">and</span> 11353_DN3_H3K4me2_regions.txt (<span class="syntax--entity syntax--name syntax--function">32967</span> total)</span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="hard-tab leading-whitespace">  </span><span class="syntax--entity syntax--name syntax--function">Comparing</span> 11353_DN3_H3K4me2_regions.txt (<span class="syntax--entity syntax--name syntax--function">32967</span> total) <span class="syntax--entity syntax--name syntax--function">and</span> 10500_DN3_H3K4me2_regions.txt (<span class="syntax--entity syntax--name syntax--function">28009</span> total)</span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="hard-tab leading-whitespace">  </span><span class="syntax--entity syntax--name syntax--function">Comparing</span> 11353_DN3_H3K4me2_regions.txt (<span class="syntax--entity syntax--name syntax--function">32967</span> total) <span class="syntax--entity syntax--name syntax--function">and</span> 11353_DN3_H3K4me2_regions.txt (<span class="syntax--entity syntax--name syntax--function">32967</span> total)</span></span>
<span class=""></span> 
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">10500_DN3_H3K4me2_regions.txt</span><span class="hard-tab"> </span>11353_DN3_H3K4me2_regions.txt<span class="hard-tab"> </span>Total<span class="hard-tab"> </span>Name</span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="hard-tab leading-whitespace">  </span><span class="syntax--entity syntax--name syntax--function">X</span><span class="hard-tab"> </span>8823<span class="hard-tab">  </span>11353_DN3_H3K4me2_regions.txt</span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">X</span><span class="hard-tab"> </span><span class="hard-tab">  </span>3223<span class="hard-tab">  </span>10500_DN3_H3K4me2_regions.txt</span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">X</span><span class="hard-tab"> </span>X<span class="hard-tab"> </span>24031<span class="hard-tab"> </span>10500_DN3_H3K4me2_regions.txt<span class="syntax--keyword syntax--operator">|</span><span class="syntax--entity syntax--name syntax--function">11353_DN3_H3K4me2_regions.txt</span></span></span></pre>
<p>Rename the overlapping peak files to something more convenient.</p>
<p>Collect all peak-files in one folder.</p>
<p><strong>3.</strong> <strong>Annotation of peak files:</strong></p>
<pre class="editor-colors lang-bash"><span class=""><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">annotatePeaks.pl</span></span></span></pre>
<p>This is Homer’s workhorse so take some time to understand the options:
<a href="http://homer.ucsd.edu/homer/ngs/annotation.html">http://homer.ucsd.edu/homer/ngs/annotation.html</a> and
<a href="http://homer.ucsd.edu/homer/ngs/quantification.html">http://homer.ucsd.edu/homer/ngs/quantification.html</a>
For now we will just use the basic proximity based annotation. This requires a built-in Homer database (needs to be separately installed). In this case we will use <em>mm10</em>.</p>
<pre class="editor-colors lang-bash"><span><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#! /bin/bash</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#SBATCH -A lu2018-2-44</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#SBATCH -p lu</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#SBATCH --reservation=lu2018-2-44</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#SBATCH -n 4 # how many processor cores to use</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#SBATCH -N 1 # how many processors to use (always use 1 here unless you know what you are doing)</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#SBATCH -t 00:20:00 # kill the job after ths hh::mm::ss time</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#SBATCH -J 'annotation_mm10' # name of the job</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#SBATCH -o 'annotation_mm10%j.out' # stdout log file</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#SBATCH -e 'annotation_mm10%j.err' # stderr log file</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#SBATCH -p dell # which partition to use</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block"># A script that preprocesses the ChIP samples.</span></span></span>
<span class=""></span> 
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#Load Homer module</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">module</span> load GCC/7.3.0-2.30 homer/4.10</span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#Basic peak-filea annotation.</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">annotatePeaks.pl</span> 11307_DN1_PU1_peaks_15.txt mm10 <span class="syntax--keyword syntax--operator">&gt;</span> 11307_DN1_PU1_peaks_15.mm10_annotated.txt</span></span>
<span class=""></span> 
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">annotatePeaks.pl</span> 11735_DN2b_PU1_peaks_15.txt mm10 <span class="syntax--keyword syntax--operator">&gt;</span> 11735_DN2b_PU1_peaks_15.mm10_annotated.txt</span></span>
<span class=""></span> 
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">annotatePeaks.pl</span> DN1_H3K4me2_regions.overlapping.txt mm10 <span class="syntax--keyword syntax--operator">&gt;</span> DN1_H3K4me2_regions.overlapping.mm10_annotated.txt</span></span>
<span class=""></span> 
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">annotatePeaks.pl</span> DN3_H3K4me2_regions.overlapping.txt mm10 <span class="syntax--keyword syntax--operator">&gt;</span> DN3_H3K4me2_regions.overlapping.mm10_annotated.txt</span></span>
<span class=""></span> 
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">module</span> purge</span></span></pre>
<p>Copy the files to your own computer, open and look at them. Be warned, if you open the file in Excel some gene names will change to dates.</p>
<p><strong>Questions to think about/homework:</strong></p>
<ul>
<li>What are the ratios of the genomic regions bound? Create a pie-chart.</li>
<li>What’s the average peak-distance to a transcription-start-site (TSS)?</li>
</ul>
<p><strong>4.</strong> <strong>Motif analysis</strong></p>
<p>When we have the peak file, basic motif-analysis in Homer is easy. Motif analysis is the reason Homer was developed and it is therefore very good at it with manually curated background models. To get all the options for basic motif-analysis, type:
<code style="font-family: Menlo, Consolas, &quot;DejaVu Sans Mono&quot;, monospace;">findMotifsGenome.pl</code></p>
<p><strong>Task:</strong> Perform motif analysis on the PU.1 ChIP-files. <strong>WARNING!</strong> A motif analysis can take a considerable amount of time if you have many peaks. Allocate accordingly.</p>
<pre class="editor-colors lang-bash"><span><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#! /bin/bash</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#SBATCH -A lu2018-2-44</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#SBATCH -p lu</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#SBATCH --reservation=lu2018-2-44</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#SBATCH -n 4 # how many processor cores to use</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#SBATCH -N 1 # how many processors to use (always use 1 here unless you know what you are doing)</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#SBATCH -t 00:30:00 # kill the job after ths hh::mm::ss time</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#SBATCH -J 'findMotifsGenome_mm10' # name of the job</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#SBATCH -o 'findMotifsGenome_mm10%j.out' # stdout log file</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#SBATCH -e 'findMotifsGenome_mm10%j.err' # stderr log file</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#SBATCH -p dell # which partition to use</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block"># A script that preprocesses the ChIP samples.</span></span></span>
<span class=""></span> 
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#Load Homer module</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">module</span> load GCC/7.3.0-2.30 homer/4.10</span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#Basic peak-filea annotation.</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">findMotifsGenome.pl</span> 11307_DN1_PU1_peaks_15.mm10_annotated.txt mm10 ./Motifs-11307_DN1_PU1_peaks_15 <span class="syntax--entity syntax--other syntax--attribute-name">-size</span> 200 <span class="syntax--entity syntax--other syntax--attribute-name">-mask</span> <span class="syntax--entity syntax--other syntax--attribute-name">-p</span> 4 <span class="syntax--entity syntax--other syntax--attribute-name">-preparsedDir</span> /home/jonun/NAS2/ChIP-ATAC-workshop-2019-course-material/homer_background_preparsed_dir</span></span>
<span class=""></span> 
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">findMotifsGenome.pl</span> 11735_DN2b_PU1_peaks_15.mm10_annotated.txt mm10 ./Motifs-11735_DN2b_PU1_peaks_15 <span class="syntax--entity syntax--other syntax--attribute-name">-size</span> 200 <span class="syntax--entity syntax--other syntax--attribute-name">-mask</span> <span class="syntax--entity syntax--other syntax--attribute-name">-p</span> 4 <span class="syntax--entity syntax--other syntax--attribute-name">-preparsedDir</span> /home/jonun/NAS2/ChIP-ATAC-workshop-2019-course-material/homer_background_preparsed_dir</span></span>
<span class=""></span> 
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">module</span> purge</span></span></pre>
<p>The <code style="font-family: Menlo, Consolas, &quot;DejaVu Sans Mono&quot;, monospace;">-preparsedDir</code> option is needed since we do not have write permissions in the Homer genome dir we need to specify where Homer builds the background model.</p>
<p><strong>Open the motif-folders and compare knownResults.html and homerResults.html. What is the difference and why?</strong></p>
<p><strong>What is the most enriched motif?</strong></p>
<p><em>Motif analysis will take awhile so let’s move on while we wait.</em></p>
<p>Another very common tool-suite is MEME. This can be run online or via the command-line. There are many tools in the meme suite but we will use <strong>meme-chip</strong> which is designed for ChIP data with thousands of sequences.</p>
<p>The input for meme is fasta-formatted sequences. This can be created from the peak file. MEME also works best when the sequences have the same length. Homer’s peak size-estimate can be found in the tagInfo.txt-file in the tag directory.</p>
<blockquote>
<p>Let’s use 200 bp as the estimate.</p>
</blockquote>
<p>Load the Homer module:</p>
<pre class="editor-colors lang-bash"><span class=""><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">module</span> load GCC/7.3.0-2.30 homer/4.10</span></span></pre>
<p>Adjust the PU.1 peak files to 200 bp:</p>
<pre class="editor-colors lang-bash"><span><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">adjustPeakFile.pl</span> 11307_DN1_PU1_peaks_15.txt <span class="syntax--entity syntax--other syntax--attribute-name">-size</span> 200 <span class="syntax--keyword syntax--operator">&gt;</span> 11307_DN1_PU1_peaks_15_200.txt</span></span>
<span class=""></span> 
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">adjustPeakFile.pl</span> 11735_DN2b_PU1_peaks_15.txt <span class="syntax--entity syntax--other syntax--attribute-name">-size</span> 200 <span class="syntax--keyword syntax--operator">&gt;</span> 11735_DN2b_PU1_peaks_15_200.txt</span></span></pre>
<p><code style="font-family: Menlo, Consolas, &quot;DejaVu Sans Mono&quot;, monospace;">homeTools extract</code> can convert a peak file to a fasta-file.</p>
<pre class="editor-colors lang-bash"><span><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">homerTools</span> extract 11307_DN1_PU1_peaks_15_200.txt /projects/fs1/common/genome/lunarc/genomes/mouse/mm10/mm10.fa <span class="syntax--entity syntax--other syntax--attribute-name">-fa</span> <span class="syntax--keyword syntax--operator">&gt;</span> 11307_DN1_PU1_peaks_15_200.fa</span></span>
<span class=""></span> 
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">homerTools</span> extract 11735_DN2b_PU1_peaks_15_200.txt /projects/fs1/common/genome/lunarc/genomes/mouse/mm10/mm10.fa <span class="syntax--entity syntax--other syntax--attribute-name">-fa</span> <span class="syntax--keyword syntax--operator">&gt;</span> 11735_DN2b_PU1_peaks_15_200.fa</span></span></pre>
<p><strong>NOTE:</strong> This can also be done with bedtools getfasta after converting the peak file to a bed file but more on that later.</p>
<p>Create and run a batch-script that calls chip-meme.</p>
<pre class="editor-colors lang-bash"><span><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#! /bin/bash</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#SBATCH -A lu2018-2-44</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#SBATCH -p lu</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#SBATCH --reservation=lu2018-2-44</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#SBATCH -n 1 # how many processor cores to use</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#SBATCH -N 1 # how many processors to use (always use 1 here unless you know what you are doing)</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#SBATCH -t 02:00:00 # kill the job after ths hh::mm::ss time</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#SBATCH -J 'chip_meme' # name of the job</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#SBATCH -o 'chip_meme_mm10%j.out' # stdout log file</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#SBATCH -e 'chip_meme%j.err' # stderr log file</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#SBATCH -p dell # which partition to use</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block"># A script that preprocesses the ChIP samples.</span></span></span>
<span class=""></span> 
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#Load meme module</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">module</span> load GCC/7.3.0-2.30  OpenMPI/3.1.1 MEME/5.0.4</span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#Run meme-chip on the fasta files</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">meme-chip</span> <span class="syntax--entity syntax--other syntax--attribute-name">-db</span> /projects/fs3/jonun/ChIP-ATAC-workshop-2019-course-material/meme_motif_databases/JASPAR/JASPAR2018_CORE_vertebrates_non-redundant.meme <span class="syntax--entity syntax--other syntax--attribute-name">-meme-nmotifs</span> 10 <span class="syntax--entity syntax--other syntax--attribute-name">-o</span> meme_11735_DN2b_PU1_peaks_15_200  11735_DN2b_PU1_peaks_15_200.fa</span></span>
<span class=""></span> 
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">meme-chip</span> <span class="syntax--entity syntax--other syntax--attribute-name">-db</span> /projects/fs3/jonun/ChIP-ATAC-workshop-2019-course-material/meme_motif_databases/JASPAR/JASPAR2018_CORE_vertebrates_non-redundant.meme <span class="syntax--entity syntax--other syntax--attribute-name">-meme-nmotifs</span> 10 <span class="syntax--entity syntax--other syntax--attribute-name">-o</span> meme_11307_DN1_PU1_peaks_15_200  11307_DN1_PU1_peaks_15_200.fa</span></span>
<span class=""></span> 
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">module</span> purge</span></span></pre>
<p><strong>OR</strong>
these .fa-files can be uploaded to the online tool:
<a href="http://meme-suite.org/tools/meme-chip">http://meme-suite.org/tools/meme-chip</a></p>
<p>Open the result-folder and study the output.</p>
<p><strong>Homework:</strong> There are many more tools in the MEME suite. Explore them and see what they can do for you!</p>
<p><a href="http://meme-suite.org/doc/overview.html?man_type=web">http://meme-suite.org/doc/overview.html?man_type=web</a></p>
<p><em>Motif analysis will take awhile so let’s move on while we wait.</em></p>
<h3>ATAC-seq assignments</h3>
<p>We have now looked at some basic ChIP-seq analysis and we will come back to that again later. One “data type” that is very similar to ChIP-seq is ATAC-seq. In fact, it is analyzed in such a similar manner that an absolute majority of the analyzes are perfomed with ChIP-seq tools. ATAC-peak calling is, however, complicated by a few facts:</p>
<ul>
<li>ATAC-seq peaks are much more varying in size than intra-sample ChIP-peaks.</li>
<li>Lack of normalization/input controls.</li>
<li>Tn5 transposome sequence bias.</li>
<li>Partly the large abundance of mitochondrial reads but correct pre-processing deals with this qute well.</li>
</ul>
<p>In theory most peak callers can be used to find ATAC-peaks though they may require several rounds of peak calling and merging of the outputs. The most widely used is MACS2: <a href="https://github.com/taoliu/MACS">https://github.com/taoliu/MACS</a></p>
<p>We will now use MACS2 to call peaks from our filtered bam ATAC-output. There are many ways to tweak ATAC-peak calling with MACS2 but we will for now stick to ENCODE’s guidelines:</p>
<p><a href="https://docs.google.com/document/d/1f0Cm4vRyDQDu0bMehHD7P7KOMxTOP-HiNoIvL1VcBt8/edit">https://docs.google.com/document/d/1f0Cm4vRyDQDu0bMehHD7P7KOMxTOP-HiNoIvL1VcBt8/edit</a></p>
<p><strong>1. MACS2 ATAC-peak calling</strong></p>
<p>From the MACS2 manual:</p>
<p><em>Here are some examples for combining --shift and --extsize:
Option 1: To find enriched cutting sites such as some DNAse-Seq datasets. In this case, all 5’ ends of sequenced reads should be extended in both direction to smooth the pileup signals. If the wanted smoothing window is 200bps, then use ‘–nomodel --shift -75 --extsize 150’.</em></p>
<p><em>Option 2: For certain nucleosome-seq data, we need to pileup the centers of nucleosomes using a half-nucleosome size for wavelet analysis (e.g. NPS algorithm). Since the DNA wrapped on nucleosome is about 147bps, this option can be used: ‘–nomodel --shift 37 --extsize 73’.</em></p>
<p><strong>Homework:</strong> Re-run the command below with option two to see how it performs against option 1.</p>
<p>First create a batch-script that perform the operation:</p>
<pre class="editor-colors lang-bash"><span><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#! /bin/bash</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#SBATCH -A lu2018-2-44</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#SBATCH -p lu</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#SBATCH --reservation=lu2018-2-44</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#SBATCH -n 1 # how many processor cores to use</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#SBATCH -N 1 # how many processors to use (always use 1 here unless you know what you are doing)</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#SBATCH -t 00:30:00 # kill the job after ths hh::mm::ss time</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#SBATCH -J 'macs2_callpeaks' # name of the job</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#SBATCH -o 'macs2_callpeaks%j.out' # stdout log file</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#SBATCH -e 'macs2_callpeaks%j.err' # stderr log file</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#SBATCH -p dell # which partition to use</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block"># A script that preprocesses the ChIP samples.</span></span></span>
<span class=""></span> 
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#Load the MACS2 module</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">module</span> load GCC/4.9.3-2.25  OpenMPI/1.10.2 icc/2015.3.187-GNU-4.9.3-2.25  impi/5.0.3.048 ifort/2015.3.187-GNU-4.9.3-2.25  impi/5.0.3.048 MACS2/2.1.0.20150731-Python-2.7.11</span></span>
<span class=""></span> 
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--variable syntax--other syntax--member">sample_path</span>=<span class="syntax--string">"/home/jonun/NAS2/ChIP-ATAC-workshop-2019-course-material"</span></span></span>
<span class=""></span> 
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#Call peaks</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">macs2</span> callpeak <span class="syntax--entity syntax--other syntax--attribute-name">-t</span> <span class="syntax--punctuation syntax--section syntax--embedded">${</span><span class="syntax--variable syntax--other syntax--member">sample_path</span><span class="syntax--punctuation syntax--section syntax--embedded">}</span>/<span class="syntax--punctuation syntax--section syntax--embedded">${</span><span class="syntax--variable syntax--other syntax--member">sample</span><span class="syntax--punctuation syntax--section syntax--embedded">}</span>/<span class="syntax--punctuation syntax--section syntax--embedded">${</span><span class="syntax--variable syntax--other syntax--member">sample</span><span class="syntax--punctuation syntax--section syntax--embedded">}</span>.filt.bam <span class="syntax--entity syntax--other syntax--attribute-name">-g</span> mm <span class="syntax--entity syntax--other syntax--attribute-name">-f</span> AUTO <span class="syntax--entity syntax--other syntax--attribute-name">-n</span> MACS2_<span class="syntax--punctuation syntax--section syntax--embedded">${</span><span class="syntax--variable syntax--other syntax--member">sample</span><span class="syntax--punctuation syntax--section syntax--embedded">}</span> <span class="syntax--entity syntax--other syntax--attribute-name">--nomodel</span> <span class="syntax--entity syntax--other syntax--attribute-name">--shift</span> <span class="syntax--entity syntax--other syntax--attribute-name">-75</span> <span class="syntax--entity syntax--other syntax--attribute-name">--extsize</span> 150 <span class="syntax--entity syntax--other syntax--attribute-name">--keep-dup</span> all <span class="syntax--entity syntax--other syntax--attribute-name">--outdir</span> <span class="syntax--punctuation syntax--section syntax--embedded">${</span><span class="syntax--variable syntax--other syntax--member">sample_path</span><span class="syntax--punctuation syntax--section syntax--embedded">}</span>/<span class="syntax--punctuation syntax--section syntax--embedded">${</span><span class="syntax--variable syntax--other syntax--member">sample</span><span class="syntax--punctuation syntax--section syntax--embedded">}</span>/MACS2_<span class="syntax--punctuation syntax--section syntax--embedded">${</span><span class="syntax--variable syntax--other syntax--member">sample</span><span class="syntax--punctuation syntax--section syntax--embedded">}</span> <span class="syntax--entity syntax--other syntax--attribute-name">-B</span> <span class="syntax--entity syntax--other syntax--attribute-name">--SPMR</span></span></span>
<span class=""></span> 
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">module</span> purge</span></span></pre>
<p>Then create script that calls the previous script for each sample:</p>
<pre class="editor-colors lang-bash"><span><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#!/bin/bash</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#A wrapper that loops over the ATAC samples and calls batch_macs2callpeaks.sh for each sample.</span></span></span>
<span class=""></span> 
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--variable syntax--other syntax--member">samples</span>=<span class="syntax--string">"ATAC-seq_ETP_T_cells_rep1 ATAC-seq_ETP_T_cells_rep2 ATAC-seq_DN3_T_cells_rep1 ATAC-seq_DN3_T_cells_rep2"</span></span></span>
<span class=""></span> 
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--keyword syntax--control">for</span> <span class="syntax--variable syntax--other syntax--member">i</span> <span class="syntax--keyword syntax--control">in</span> <span class="syntax--punctuation syntax--section syntax--embedded">${</span><span class="syntax--variable syntax--other syntax--member">samples</span><span class="syntax--punctuation syntax--section syntax--embedded">}</span>;<span class="syntax--keyword syntax--control">do</span></span></span>
<span class=""></span> 
<span class=""><span class="syntax--source syntax--shell"><span class="leading-whitespace">        </span><span class="syntax--entity syntax--name syntax--function"><span class="syntax--support syntax--function">echo</span></span> <span class="syntax--string">"sbatch --export=sample=<span class="syntax--punctuation syntax--section syntax--embedded">${</span><span class="syntax--variable syntax--other syntax--member">i</span><span class="syntax--punctuation syntax--section syntax--embedded">}</span> batch_macs2callpeaks.sh"</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="leading-whitespace">        </span><span class="syntax--entity syntax--name syntax--function">sbatch</span> <span class="syntax--entity syntax--other syntax--attribute-name">--export=sample=</span><span class="syntax--punctuation syntax--section syntax--embedded">${</span><span class="syntax--variable syntax--other syntax--member">i</span><span class="syntax--punctuation syntax--section syntax--embedded">}</span> batch_macs2callpeaks.sh</span></span>
<span class=""></span> 
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--keyword syntax--control">done</span></span></span></pre>
<p>Run this:</p>
<pre class="editor-colors lang-bash"><span class=""><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">bash</span> MACS2_wrapper.sh</span></span></pre>
<p><strong>Discussion point:</strong> It is quite common to add the --broad option to macs2 callpeaks. How would this affect the output?</p>
<p><strong>Note:</strong> This is a more efficient way to setup the peak calling compared to how we called findPeaks with Homer. What is the difference?</p>
<p><strong>2.</strong> <strong>Subtract overlapping peaks between replicates using bedtools.</strong></p>
<p>Like in the ChIP-case we want to use reproducible peaks. Again, IDR is the best practice but we are gonna work with overlapping beaks between replicats.</p>
<p>Start by copying each .narrow peak-file into a common folder.</p>
<p>Load the bedtools module.</p>
<pre class="editor-colors lang-bash"><span class=""><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">module</span> load GCC/5.4.0-2.26  OpenMPI/1.10.3 icc/2017.1.132-GCC-6.3.0-2.27 impi/2017.1.132 ifort/2017.1.132-GCC-6.3.0-2.27 impi/2017.1.132 BEDTools/2.26.0</span></span></pre>
<p>Like Homer, this is a suite that contain many usefool tools for processing of files in the bed-format and you can read about them here:
<a href="https://bedtools.readthedocs.io/en/latest/#">https://bedtools.readthedocs.io/en/latest/#</a></p>
<p>For now we are gonna use <code style="font-family: Menlo, Consolas, &quot;DejaVu Sans Mono&quot;, monospace;">bedtools intersect</code>.
Look at the options with:</p>
<pre class="editor-colors lang-bash"><span class=""><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">bedtools</span> intersect <span class="syntax--entity syntax--other syntax--attribute-name">-h</span></span></span></pre>
<pre class="editor-colors lang-bash"><span><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">bedtools</span> intersect <span class="syntax--entity syntax--other syntax--attribute-name">-a</span> MACS2_ATAC-seq_ETP_T_cells_rep1_peaks.narrowPeak <span class="syntax--entity syntax--other syntax--attribute-name">-b</span> MACS2_ATAC-seq_ETP_T_cells_rep2_peaks.narrowPeak <span class="syntax--keyword syntax--operator">&gt;</span> ATAC-seq_ETP_T_cells_rep1_2.overlapping.bed</span></span>
<span class=""></span> 
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">bedtools</span> intersect <span class="syntax--entity syntax--other syntax--attribute-name">-a</span> MACS2_ATAC-seq_DN3_T_cells_rep1_peaks.narrowPeak <span class="syntax--entity syntax--other syntax--attribute-name">-b</span> MACS2_ATAC-seq_DN3_T_cells_rep2_peaks.narrowPeak <span class="syntax--keyword syntax--operator">&gt;</span> ATAC-seq_DN3_T_cells_rep1_2.overlapping.bed</span></span></pre>
<p>How many peaks do the overlaps contain?
The number of lines in a file can be derived with the command <code style="font-family: Menlo, Consolas, &quot;DejaVu Sans Mono&quot;, monospace;">cat file | wc -l</code>.</p>
<blockquote>
<p>ETP: 56994</p>
<p>DN3: 20616</p>
</blockquote>
<p>This can of course be interpreted as there are much fewer open sites at the DN3 stage but it could also mean that the DN3 samples suffer from a lower signal-to-noise.</p>
<p><strong>3.</strong> <strong>Cell type specific peak subset analysis.</strong></p>
<p>We will now derive the peaks that are unique to ETPs and DN3, respectively, and investigate them further.</p>
<p>To only report entries in <code style="font-family: Menlo, Consolas, &quot;DejaVu Sans Mono&quot;, monospace;">-a</code> that is not in <code style="font-family: Menlo, Consolas, &quot;DejaVu Sans Mono&quot;, monospace;">-b</code> we will use the <code style="font-family: Menlo, Consolas, &quot;DejaVu Sans Mono&quot;, monospace;">-v</code> option with <code style="font-family: Menlo, Consolas, &quot;DejaVu Sans Mono&quot;, monospace;">bedtools intersect</code>.</p>
<pre class="editor-colors lang-bash"><span><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">bedtools</span> intersect <span class="syntax--entity syntax--other syntax--attribute-name">-a</span> ATAC-seq_ETP_T_cells_rep1_2.overlapping.bed <span class="syntax--entity syntax--other syntax--attribute-name">-b</span> ATAC-seq_DN3_T_cells_rep1_2.overlapping.bed <span class="syntax--entity syntax--other syntax--attribute-name">-v</span> <span class="syntax--keyword syntax--operator">&gt;</span> ETP_unique_ATAC_peaks.bed</span></span>
<span class=""></span> 
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">bedtools</span> intersect <span class="syntax--entity syntax--other syntax--attribute-name">-b</span> ATAC-seq_ETP_T_cells_rep1_2.overlapping.bed <span class="syntax--entity syntax--other syntax--attribute-name">-a</span> ATAC-seq_DN3_T_cells_rep1_2.overlapping.bed <span class="syntax--entity syntax--other syntax--attribute-name">-v</span> <span class="syntax--keyword syntax--operator">&gt;</span> DN3_unique_ATAC_peaks.bed</span></span></pre>
<p>An alternative way to do the analysis above would be to calculate statistical difference with, for instance, DESeq2 or edgeR. This would allow for detection of dynamic changes in chromatin accessibility whereas the analysis above gives a more black and white picture of peaks that are completely lost or gained between the two stages.</p>
<p>Load the Homer module and annotate these peak files.</p>
<pre class="editor-colors lang-bash"><span><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">module</span> purge</span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">module</span> load GCC/7.3.0-2.30 homer/4.10</span></span>
<span class=""></span> 
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">annotatePeaks.pl</span> ETP_unique_ATAC_peaks.bed mm10 <span class="syntax--keyword syntax--operator">&gt;</span> ETP_unique_ATAC_peaks.mm10_annotated.txt</span></span>
<span class=""></span> 
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">annotatePeaks.pl</span> DN3_unique_ATAC_peaks.bed mm10 <span class="syntax--keyword syntax--operator">&gt;</span> DN3_unique_ATAC_peaks.mm10_annotated.txt</span></span></pre>
<p><em>Open and look at the files.</em></p>
<p><strong>4.</strong> <strong>Motif analysis on subset specific ATAC-peaks.</strong></p>
<p><strong>Task:</strong> What motifs are enriched in the two different categories?</p>
<pre class="editor-colors lang-bash"><span><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#! /bin/bash</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#SBATCH -A lu2018-2-44</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#SBATCH -p lu</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#SBATCH --reservation=lu2018-2-44</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#SBATCH -n 4 # how many processor cores to use</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#SBATCH -N 1 # how many processors to use (always use 1 here unless you know what you are doing)</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#SBATCH -t 01:00:00 # kill the job after ths hh::mm::ss time</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#SBATCH -J 'findMotifsGenome_mm10' # name of the job</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#SBATCH -o 'findMotifsGenome_mm10%j.out' # stdout log file</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#SBATCH -e 'findMotifsGenome_mm10%j.err' # stderr log file</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#SBATCH -p dell # which partition to use</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block"># A script that preprocesses the ChIP samples.</span></span></span>
<span class=""></span> 
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#Load Homer module</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">module</span> load GCC/7.3.0-2.30 homer/4.10</span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#Basic peak-filea annotation.</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">findMotifsGenome.pl</span> ETP_unique_ATAC_peaks.mm10_annotated.txt mm10 ./Motifs-ETP_unique_ATAC_peaks <span class="syntax--entity syntax--other syntax--attribute-name">-size</span> 200 <span class="syntax--entity syntax--other syntax--attribute-name">-mask</span> <span class="syntax--entity syntax--other syntax--attribute-name">-p</span> 4 <span class="syntax--entity syntax--other syntax--attribute-name">-preparsedDir</span> /home/jonun/NAS2/ChIP-ATAC-workshop-2019-course-material/homer_background_preparsed_dir</span></span>
<span class=""></span> 
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">findMotifsGenome.pl</span> DN3_unique_ATAC_peaks.mm10_annotated.txt mm10 ./Motifs-DN3_unique_ATAC_peaks <span class="syntax--entity syntax--other syntax--attribute-name">-size</span> 200 <span class="syntax--entity syntax--other syntax--attribute-name">-mask</span> <span class="syntax--entity syntax--other syntax--attribute-name">-p</span> 4 <span class="syntax--entity syntax--other syntax--attribute-name">-preparsedDir</span> /home/jonun/NAS2/ChIP-ATAC-workshop-2019-course-material/homer_background_preparsed_dir</span></span>
<span class=""></span> 
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">module</span> purge</span></span></pre>
<p>Open the homerResults.html and find out <img class="emoji" draggable="false" alt="😃" src="/Users/Jonas/.atom/packages/markdown-preview-plus/node_modules/twemoji/2/svg/1f603.svg"></p>
<h3>Combined ChIP- and ATAC-seq analyses</h3>
<p><strong>1. We will now try to answer a common question. Can we identify TF binding in accessible/active and inaccessible/inactive chromatin and how do they differ?</strong></p>
<p>In our case the questions would be: <strong>How many PU.1 peaks are there in ATAC-accessible reagions? In H3K4me2 marked regions?</strong></p>
<p>These question can be answered with both Homer and bedtools intersect but for now let us use Homer’s <code style="font-family: Menlo, Consolas, &quot;DejaVu Sans Mono&quot;, monospace;">mergePeaks</code>.</p>
<p>PU.1 overlapping with H3K4me2 marked histones.</p>
<pre class="editor-colors lang-bash"><span><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">mergePeaks</span> 11307_DN1_PU1_peaks_15.txt DN1_H3K4me2_regions.overlapping.txt <span class="syntax--entity syntax--other syntax--attribute-name">-prefix</span> ETP_H3K4me2</span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="hard-tab leading-whitespace">  </span><span class="syntax--entity syntax--name syntax--function">Max</span> distance to merge: direct overlap required (<span class="syntax--entity syntax--name syntax--function">-d</span> given)</span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="hard-tab leading-whitespace">  </span><span class="syntax--entity syntax--name syntax--function">Merging</span> peaks...</span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="hard-tab leading-whitespace">  </span><span class="syntax--entity syntax--name syntax--function">Comparing</span> 11307_DN1_PU1_peaks_15.txt (<span class="syntax--entity syntax--name syntax--function">37476</span> total) <span class="syntax--entity syntax--name syntax--function">and</span> 11307_DN1_PU1_peaks_15.txt (<span class="syntax--entity syntax--name syntax--function">37476</span> total)</span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="hard-tab leading-whitespace">  </span><span class="syntax--entity syntax--name syntax--function">Comparing</span> 11307_DN1_PU1_peaks_15.txt (<span class="syntax--entity syntax--name syntax--function">37476</span> total) <span class="syntax--entity syntax--name syntax--function">and</span> DN1_H3K4me2_regions.overlapping.txt (<span class="syntax--entity syntax--name syntax--function">36998</span> total)</span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="hard-tab leading-whitespace">  </span><span class="syntax--entity syntax--name syntax--function">Comparing</span> DN1_H3K4me2_regions.overlapping.txt (<span class="syntax--entity syntax--name syntax--function">36998</span> total) <span class="syntax--entity syntax--name syntax--function">and</span> 11307_DN1_PU1_peaks_15.txt (<span class="syntax--entity syntax--name syntax--function">37476</span> total)</span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="hard-tab leading-whitespace">  </span><span class="syntax--entity syntax--name syntax--function">Comparing</span> DN1_H3K4me2_regions.overlapping.txt (<span class="syntax--entity syntax--name syntax--function">36998</span> total) <span class="syntax--entity syntax--name syntax--function">and</span> DN1_H3K4me2_regions.overlapping.txt (<span class="syntax--entity syntax--name syntax--function">36998</span> total)</span></span>
<span class=""></span> 
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">11307_DN1_PU1_peaks_15.txt</span><span class="hard-tab">  </span>DN1_H3K4me2_regions.overlapping.txt<span class="hard-tab"> </span>Total<span class="hard-tab"> </span>Name</span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="hard-tab leading-whitespace">  </span><span class="syntax--entity syntax--name syntax--function">X</span><span class="hard-tab"> </span>18031<span class="hard-tab"> </span>DN1_H3K4me2_regions.overlapping.txt</span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">X</span><span class="hard-tab"> </span><span class="hard-tab">  </span>13367<span class="hard-tab"> </span>11307_DN1_PU1_peaks_15.txt</span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">X</span><span class="hard-tab"> </span>X<span class="hard-tab"> </span>18967<span class="hard-tab"> </span>11307_DN1_PU1_peaks_15.txt<span class="syntax--keyword syntax--operator">|</span><span class="syntax--entity syntax--name syntax--function">DN1_H3K4me2_regions.overlapping.txt</span></span></span></pre>
<pre class="editor-colors lang-bash"><span><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">mergePeaks</span> 11735_DN2b_PU1_peaks_15.txt DN3_H3K4me2_regions.overlapping.txt <span class="syntax--entity syntax--other syntax--attribute-name">-prefix</span> DN3_H3K4me2</span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="hard-tab leading-whitespace">  </span><span class="syntax--entity syntax--name syntax--function">Max</span> distance to merge: direct overlap required (<span class="syntax--entity syntax--name syntax--function">-d</span> given)</span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="hard-tab leading-whitespace">  </span><span class="syntax--entity syntax--name syntax--function">Merging</span> peaks...</span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="hard-tab leading-whitespace">  </span><span class="syntax--entity syntax--name syntax--function">Comparing</span> 11735_DN2b_PU1_peaks_15.txt (<span class="syntax--entity syntax--name syntax--function">6688</span> total) <span class="syntax--entity syntax--name syntax--function">and</span> 11735_DN2b_PU1_peaks_15.txt (<span class="syntax--entity syntax--name syntax--function">6688</span> total)</span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="hard-tab leading-whitespace">  </span><span class="syntax--entity syntax--name syntax--function">Comparing</span> 11735_DN2b_PU1_peaks_15.txt (<span class="syntax--entity syntax--name syntax--function">6688</span> total) <span class="syntax--entity syntax--name syntax--function">and</span> DN3_H3K4me2_regions.overlapping.txt (<span class="syntax--entity syntax--name syntax--function">24031</span> total)</span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="hard-tab leading-whitespace">  </span><span class="syntax--entity syntax--name syntax--function">Comparing</span> DN3_H3K4me2_regions.overlapping.txt (<span class="syntax--entity syntax--name syntax--function">24031</span> total) <span class="syntax--entity syntax--name syntax--function">and</span> 11735_DN2b_PU1_peaks_15.txt (<span class="syntax--entity syntax--name syntax--function">6688</span> total)</span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="hard-tab leading-whitespace">  </span><span class="syntax--entity syntax--name syntax--function">Comparing</span> DN3_H3K4me2_regions.overlapping.txt (<span class="syntax--entity syntax--name syntax--function">24031</span> total) <span class="syntax--entity syntax--name syntax--function">and</span> DN3_H3K4me2_regions.overlapping.txt (<span class="syntax--entity syntax--name syntax--function">24031</span> total)</span></span>
<span class=""></span> 
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">11735_DN2b_PU1_peaks_15.txt</span><span class="hard-tab"> </span>DN3_H3K4me2_regions.overlapping.txt<span class="hard-tab"> </span>Total<span class="hard-tab"> </span>Name</span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="hard-tab leading-whitespace">  </span><span class="syntax--entity syntax--name syntax--function">X</span><span class="hard-tab"> </span>20729<span class="hard-tab"> </span>DN3_H3K4me2_regions.overlapping.txt</span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">X</span><span class="hard-tab"> </span><span class="hard-tab">  </span>3175<span class="hard-tab">  </span>11735_DN2b_PU1_peaks_15.txt</span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">X</span><span class="hard-tab"> </span>X<span class="hard-tab"> </span>3302<span class="hard-tab">  </span>11735_DN2b_PU1_peaks_15.txt<span class="syntax--keyword syntax--operator">|</span><span class="syntax--entity syntax--name syntax--function">DN3_H3K4me2_regions.overlapping.txt</span></span></span></pre>
<p>PU.1 overlapping with ATAC-accessible chromatin.</p>
<pre class="editor-colors lang-bash"><span><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">mergePeaks</span> 11307_DN1_PU1_peaks_15.txt ../MACS2_peak_files/ATAC-seq_ETP_T_cells_rep1_2.overlapping.bed <span class="syntax--entity syntax--other syntax--attribute-name">-prefix</span> ETP_ATAC</span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="hard-tab leading-whitespace">  </span><span class="syntax--entity syntax--name syntax--function">Max</span> distance to merge: direct overlap required (<span class="syntax--entity syntax--name syntax--function">-d</span> given)</span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="hard-tab leading-whitespace">  </span><span class="syntax--entity syntax--name syntax--function">Merging</span> peaks...</span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="hard-tab leading-whitespace">  </span><span class="syntax--entity syntax--name syntax--function">Duplicate</span> peak name (<span class="syntax--entity syntax--name syntax--function">MACS2_ATAC-seq_ETP_T_cells_rep1_peak_40</span>) - <span class="syntax--entity syntax--name syntax--function">this</span> could potentially cause problems</span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="hard-tab leading-whitespace">  </span><span class="hard-tab leading-whitespace">  </span><span class="syntax--entity syntax--name syntax--function">Sometimes</span> unavoidable for BED/2DBED formats</span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="hard-tab leading-whitespace">  </span><span class="hard-tab leading-whitespace">  </span><span class="syntax--entity syntax--name syntax--function">New</span> name for this peak is MACS2_ATAC-seq_ETP_T_cells_rep1_peak_40--2</span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="hard-tab leading-whitespace">  </span><span class="hard-tab leading-whitespace">  </span><span class="syntax--entity syntax--name syntax--function">Warning</span> over 1000 peaks with duplicate names</span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="hard-tab leading-whitespace">  </span><span class="syntax--entity syntax--name syntax--function">Comparing</span> 11307_DN1_PU1_peaks_15.txt (<span class="syntax--entity syntax--name syntax--function">37476</span> total) <span class="syntax--entity syntax--name syntax--function">and</span> 11307_DN1_PU1_peaks_15.txt (<span class="syntax--entity syntax--name syntax--function">37476</span> total)</span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="hard-tab leading-whitespace">  </span><span class="syntax--entity syntax--name syntax--function">Comparing</span> 11307_DN1_PU1_peaks_15.txt (<span class="syntax--entity syntax--name syntax--function">37476</span> total) <span class="syntax--entity syntax--name syntax--function">and</span> ../MACS2_peak_files/ATAC-seq_ETP_T_cells_rep1_2.overlapping.bed (<span class="syntax--entity syntax--name syntax--function">56994</span> total)</span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="hard-tab leading-whitespace">  </span><span class="syntax--entity syntax--name syntax--function">Comparing</span> ../MACS2_peak_files/ATAC-seq_ETP_T_cells_rep1_2.overlapping.bed (<span class="syntax--entity syntax--name syntax--function">56994</span> total) <span class="syntax--entity syntax--name syntax--function">and</span> 11307_DN1_PU1_peaks_15.txt (<span class="syntax--entity syntax--name syntax--function">37476</span> total)</span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="hard-tab leading-whitespace">  </span><span class="syntax--entity syntax--name syntax--function">Comparing</span> ../MACS2_peak_files/ATAC-seq_ETP_T_cells_rep1_2.overlapping.bed (<span class="syntax--entity syntax--name syntax--function">56994</span> total) <span class="syntax--entity syntax--name syntax--function">and</span> ../MACS2_peak_files/ATAC-seq_ETP_T_cells_rep1_2.overlapping.bed (<span class="syntax--entity syntax--name syntax--function">56994</span> total)</span></span>
<span class=""></span> 
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">11307_DN1_PU1_peaks_15.txt</span><span class="hard-tab">  </span>../MACS2_peak_files/ATAC-seq_ETP_T_cells_rep1_2.overlapping.bed<span class="hard-tab"> </span>Total<span class="hard-tab"> </span>Name</span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="hard-tab leading-whitespace">  </span><span class="syntax--entity syntax--name syntax--function">X</span><span class="hard-tab"> </span>35479<span class="hard-tab"> </span>../MACS2_peak_files/ATAC-seq_ETP_T_cells_rep1_2.overlapping.bed</span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">X</span><span class="hard-tab"> </span><span class="hard-tab">  </span>15783<span class="hard-tab"> </span>11307_DN1_PU1_peaks_15.txt</span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">X</span><span class="hard-tab"> </span>X<span class="hard-tab"> </span>21458<span class="hard-tab"> </span>11307_DN1_PU1_peaks_15.txt<span class="syntax--keyword syntax--operator">|</span><span class="syntax--entity syntax--name syntax--function">../MACS2_peak_files/ATAC-seq_ETP_T_cells_rep1_2.overlapping.bed</span></span></span></pre>
<pre class="editor-colors lang-bash"><span><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">mergePeaks</span> 11735_DN2b_PU1_peaks_15.txt ../MACS2_peak_files/ATAC-seq_DN3_T_cells_rep1_2.overlapping.bed <span class="syntax--entity syntax--other syntax--attribute-name">-prefix</span> DN3_ATAC</span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="hard-tab leading-whitespace">  </span><span class="syntax--entity syntax--name syntax--function">Max</span> distance to merge: direct overlap required (<span class="syntax--entity syntax--name syntax--function">-d</span> given)</span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="hard-tab leading-whitespace">  </span><span class="syntax--entity syntax--name syntax--function">Merging</span> peaks...</span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="hard-tab leading-whitespace">  </span><span class="syntax--entity syntax--name syntax--function">Duplicate</span> peak name (<span class="syntax--entity syntax--name syntax--function">MACS2_ATAC-seq_DN3_T_cells_rep1_peak_10</span>) - <span class="syntax--entity syntax--name syntax--function">this</span> could potentially cause problems</span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="hard-tab leading-whitespace">  </span><span class="hard-tab leading-whitespace">  </span><span class="syntax--entity syntax--name syntax--function">Sometimes</span> unavoidable for BED/2DBED formats</span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="hard-tab leading-whitespace">  </span><span class="hard-tab leading-whitespace">  </span><span class="syntax--entity syntax--name syntax--function">New</span> name for this peak is MACS2_ATAC-seq_DN3_T_cells_rep1_peak_10--2</span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="hard-tab leading-whitespace">  </span><span class="syntax--entity syntax--name syntax--function">Comparing</span> 11735_DN2b_PU1_peaks_15.txt (<span class="syntax--entity syntax--name syntax--function">6688</span> total) <span class="syntax--entity syntax--name syntax--function">and</span> 11735_DN2b_PU1_peaks_15.txt (<span class="syntax--entity syntax--name syntax--function">6688</span> total)</span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="hard-tab leading-whitespace">  </span><span class="syntax--entity syntax--name syntax--function">Comparing</span> 11735_DN2b_PU1_peaks_15.txt (<span class="syntax--entity syntax--name syntax--function">6688</span> total) <span class="syntax--entity syntax--name syntax--function">and</span> ../MACS2_peak_files/ATAC-seq_DN3_T_cells_rep1_2.overlapping.bed (<span class="syntax--entity syntax--name syntax--function">20616</span> total)</span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="hard-tab leading-whitespace">  </span><span class="syntax--entity syntax--name syntax--function">Comparing</span> ../MACS2_peak_files/ATAC-seq_DN3_T_cells_rep1_2.overlapping.bed (<span class="syntax--entity syntax--name syntax--function">20616</span> total) <span class="syntax--entity syntax--name syntax--function">and</span> 11735_DN2b_PU1_peaks_15.txt (<span class="syntax--entity syntax--name syntax--function">6688</span> total)</span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="hard-tab leading-whitespace">  </span><span class="syntax--entity syntax--name syntax--function">Comparing</span> ../MACS2_peak_files/ATAC-seq_DN3_T_cells_rep1_2.overlapping.bed (<span class="syntax--entity syntax--name syntax--function">20616</span> total) <span class="syntax--entity syntax--name syntax--function">and</span> ../MACS2_peak_files/ATAC-seq_DN3_T_cells_rep1_2.overlapping.bed (<span class="syntax--entity syntax--name syntax--function">20616</span> total)</span></span>
<span class=""></span> 
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">11735_DN2b_PU1_peaks_15.txt</span><span class="hard-tab"> </span>../MACS2_peak_files/ATAC-seq_DN3_T_cells_rep1_2.overlapping.bed<span class="hard-tab"> </span>Total<span class="hard-tab"> </span>Name</span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="hard-tab leading-whitespace">  </span><span class="syntax--entity syntax--name syntax--function">X</span><span class="hard-tab"> </span>18329<span class="hard-tab"> </span>../MACS2_peak_files/ATAC-seq_DN3_T_cells_rep1_2.overlapping.bed</span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">X</span><span class="hard-tab"> </span><span class="hard-tab">  </span>4398<span class="hard-tab">  </span>11735_DN2b_PU1_peaks_15.txt</span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">X</span><span class="hard-tab"> </span>X<span class="hard-tab"> </span>2284<span class="hard-tab">  </span>11735_DN2b_PU1_peaks_15.txt<span class="syntax--keyword syntax--operator">|</span><span class="syntax--entity syntax--name syntax--function">../MACS2_peak_files/ATAC-seq_DN3_T_cells_rep1_2.overlapping.bed</span></span></span></pre>
<p>These output files can now be used for downstream analysis to answer, for instance, if PU.1 binding to promoters are enriched in open and active chromatin or can we identify co-enrichment of other motifs in a specific category etc?</p>
<p><strong>2. Use deepTools to plot the H3K4me2 and ATAC fragment distribution around PU.1 peaks.</strong></p>
<p>In this part we will plot the read distribution of PU.1, H3K4me2 and ATAC, around a combined list of DN1 + DN2b PU.1 peaks.</p>
<p><a href="https://deeptools.readthedocs.io/en/develop/content/list_of_tools.html">https://deeptools.readthedocs.io/en/develop/content/list_of_tools.html</a></p>
<p>This is a multistep rocket:</p>
<ul>
<li>Start by combining the DN1 (ETP) and DN3 PU.1 peak files:</li>
</ul>
<pre class="editor-colors lang-bash"><span><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">module</span> load GCC/7.3.0-2.30 homer/4.10</span></span>
<span class=""></span> 
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">mergePeaks</span> 11307_DN1_PU1_peaks_15.txt 11735_DN2b_PU1_peaks_15.txt <span class="syntax--keyword syntax--operator">&gt;</span> PU1_peaks.txt</span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#Turn this into a bed-file.</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">pos2bed.pl</span> PU1_peaks.txt <span class="syntax--keyword syntax--operator">&gt;</span> PU1_peaks.bed</span></span>
<span class=""></span> 
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">module</span> purge</span></span></pre>
<ul>
<li>Then we will create the files (bigwigs) to be used as the input for the matrix to be visualized. <strong>BONUS:</strong> These bigwig tracks can also be visulized in a genome browser, like IGV or UCSC which is something I <strong>ALWAYS</strong> would recommend. This is done with the bamCoverage function in deepTools but first, load the deepTools module.</li>
</ul>
<pre class="editor-colors lang-bash"><span class=""><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">module</span> load icc/2017.4.196-GCC-6.4.0-2.28  impi/2017.3.196 ifort/2017.4.196-GCC-6.4.0-2.28  impi/2017.3.196 deepTools/2.5.4-Python-3.6.3</span></span></pre>
<pre class="editor-colors lang-bash"><span class=""><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">bamCoverage</span> <span class="syntax--entity syntax--other syntax--attribute-name">-h</span></span></span></pre>
<p>Create a sbatch script for this task.</p>
<pre class="editor-colors lang-bash"><span><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#! /bin/bash</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#SBATCH -A lu2018-2-44</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#SBATCH -p lu</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#SBATCH --reservation=lu2018-2-44</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#SBATCH -n 4 # how many processor cores to use</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#SBATCH -N 1 # how many processors to use (always use 1 here unless you know what you are doing)</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#SBATCH -t 00:20:00 # kill the job after ths hh::mm::ss time</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#SBATCH -J 'bamCoverage' # name of the job</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#SBATCH -o 'bamCoverage%j.out' # stdout log file</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#SBATCH -e 'bamCoverage%j.err' # stderr log file</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#SBATCH -p dell # which partition to use</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block"># A script that preprocesses the ChIP samples.</span></span></span>
<span class=""></span> 
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--variable syntax--other syntax--member">sample_path</span>=<span class="syntax--string">"/home/jonun/NAS2/ChIP-ATAC-workshop-2019-course-material"</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#genome_size=2652783500 #mm10 genome size non-N bases</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">module</span> load GCC/6.4.0-2.28  OpenMPI/2.1.2 SAMtools/1.7</span></span>
<span class=""></span> 
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#First we must index the bam-file with samtools.</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">samtools</span> index <span class="syntax--punctuation syntax--section syntax--embedded">${</span><span class="syntax--variable syntax--other syntax--member">sample_path</span><span class="syntax--punctuation syntax--section syntax--embedded">}</span>/<span class="syntax--punctuation syntax--section syntax--embedded">${</span><span class="syntax--variable syntax--other syntax--member">sample</span><span class="syntax--punctuation syntax--section syntax--embedded">}</span>/<span class="syntax--punctuation syntax--section syntax--embedded">${</span><span class="syntax--variable syntax--other syntax--member">sample</span><span class="syntax--punctuation syntax--section syntax--embedded">}</span>.filt.bam</span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">module</span> purge</span></span>
<span class=""></span> 
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#Load the deepTools module</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">module</span> load GCC/7.3.0-2.30 OpenMPI/3.1.1</span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">module</span> load deepTools/2.5.4-Python-3.6.6</span></span>
<span class=""></span> 
<span class=""></span> 
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#Make bigwigs.</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">bamCoverage</span> <span class="syntax--entity syntax--other syntax--attribute-name">-b</span> <span class="syntax--punctuation syntax--section syntax--embedded">${</span><span class="syntax--variable syntax--other syntax--member">sample_path</span><span class="syntax--punctuation syntax--section syntax--embedded">}</span>/<span class="syntax--punctuation syntax--section syntax--embedded">${</span><span class="syntax--variable syntax--other syntax--member">sample</span><span class="syntax--punctuation syntax--section syntax--embedded">}</span>/<span class="syntax--punctuation syntax--section syntax--embedded">${</span><span class="syntax--variable syntax--other syntax--member">sample</span><span class="syntax--punctuation syntax--section syntax--embedded">}</span>.filt.bam  <span class="syntax--entity syntax--other syntax--attribute-name">--normalizeUsingRPKM</span> <span class="syntax--entity syntax--other syntax--attribute-name">-p</span> 4 <span class="syntax--entity syntax--other syntax--attribute-name">--centerReads</span> <span class="syntax--entity syntax--other syntax--attribute-name">-o</span> <span class="syntax--punctuation syntax--section syntax--embedded">${</span><span class="syntax--variable syntax--other syntax--member">sample_path</span><span class="syntax--punctuation syntax--section syntax--embedded">}</span>/<span class="syntax--punctuation syntax--section syntax--embedded">${</span><span class="syntax--variable syntax--other syntax--member">sample</span><span class="syntax--punctuation syntax--section syntax--embedded">}</span>/<span class="syntax--punctuation syntax--section syntax--embedded">${</span><span class="syntax--variable syntax--other syntax--member">sample</span><span class="syntax--punctuation syntax--section syntax--embedded">}</span>.bw</span></span>
<span class=""></span> 
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">module</span> purge</span></span></pre>
<p>Create a wrapper that loops through the files (to save time we select one of the replicates).</p>
<pre class="editor-colors lang-bash"><span><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#!/bin/bash</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#A wrapper that loops over the samples and calls batch_bamCoverage.sh for each sample.</span></span></span>
<span class=""></span> 
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--variable syntax--other syntax--member">samples</span>=<span class="syntax--string">"11307_DN1_PU1 11735_DN2b_PU1 10340_DN1_H3K4me2 10500_DN3_H3K4me2 ATAC-seq_ETP_T_cells_rep1 ATAC-seq_DN3_T_cells_rep1"</span></span></span>
<span class=""></span> 
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--keyword syntax--control">for</span> <span class="syntax--variable syntax--other syntax--member">i</span> <span class="syntax--keyword syntax--control">in</span> <span class="syntax--punctuation syntax--section syntax--embedded">${</span><span class="syntax--variable syntax--other syntax--member">samples</span><span class="syntax--punctuation syntax--section syntax--embedded">}</span>;<span class="syntax--keyword syntax--control">do</span></span></span>
<span class=""></span> 
<span class=""><span class="syntax--source syntax--shell"><span class="leading-whitespace">        </span><span class="syntax--entity syntax--name syntax--function"><span class="syntax--support syntax--function">echo</span></span> <span class="syntax--string">"sbatch --export=sample=<span class="syntax--punctuation syntax--section syntax--embedded">${</span><span class="syntax--variable syntax--other syntax--member">i</span><span class="syntax--punctuation syntax--section syntax--embedded">}</span> batch_bamCoverage.sh"</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="leading-whitespace">        </span><span class="syntax--entity syntax--name syntax--function">sbatch</span> <span class="syntax--entity syntax--other syntax--attribute-name">--export=sample=</span><span class="syntax--punctuation syntax--section syntax--embedded">${</span><span class="syntax--variable syntax--other syntax--member">i</span><span class="syntax--punctuation syntax--section syntax--embedded">}</span> batch_bamCoverage.sh</span></span>
<span class=""></span> 
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--keyword syntax--control">done</span></span></span></pre>
<p>Run this:</p>
<pre class="editor-colors lang-bash"><span class=""><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">bash</span> bamCoverage_wrapper.sh</span></span></pre>
<p>Great! Now we have created us some bigwig files to compute a matrix from with computeMatrix.
Write a sbatch script for this task.</p>
<pre class="editor-colors lang-bash"><span><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#! /bin/bash</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#SBATCH -A lu2018-2-44</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#SBATCH -p lu</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#SBATCH --reservation=lu2018-2-44</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#SBATCH -n 1 # how many processor cores to use</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#SBATCH -N 1 # how many processors to use (always use 1 here unless you know what you are doing)</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#SBATCH -t 00:30:00 # kill the job after ths hh::mm::ss time</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#SBATCH -J 'computeMatrix' # name of the job</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#SBATCH -o 'computeMatrix%j.out' # stdout log file</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#SBATCH -e 'computeMatrix%j.err' # stderr log file</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#SBATCH -p dell # which partition to use</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block"># A script that preprocesses the ChIP samples.</span></span></span>
<span class=""></span> 
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#Load the deepTools module</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">module</span> load GCC/7.3.0-2.30 OpenMPI/3.1.1</span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">module</span> load deepTools/2.5.4-Python-3.6.6</span></span>
<span class=""></span> 
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--variable syntax--other syntax--member">sample_path</span>=<span class="syntax--string">"/projects/fs3/jonun/ChIP-ATAC-workshop-2019-course-material"</span></span></span>
<span class=""></span> 
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#Make matrix.</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">computeMatrix</span> reference-point <span class="syntax--entity syntax--other syntax--attribute-name">-S</span> <span class="syntax--punctuation syntax--section syntax--embedded">${</span><span class="syntax--variable syntax--other syntax--member">sample_path</span><span class="syntax--punctuation syntax--section syntax--embedded">}</span>/11307_DN1_PU1/11307_DN1_PU1.bw <span class="syntax--punctuation syntax--section syntax--embedded">${</span><span class="syntax--variable syntax--other syntax--member">sample_path</span><span class="syntax--punctuation syntax--section syntax--embedded">}</span>/11735_DN2b_PU1/11735_DN2b_PU1.bw <span class="syntax--punctuation syntax--section syntax--embedded">${</span><span class="syntax--variable syntax--other syntax--member">sample_path</span><span class="syntax--punctuation syntax--section syntax--embedded">}</span>/10340_DN1_H3K4me2/10340_DN1_H3K4me2.bw <span class="syntax--punctuation syntax--section syntax--embedded">${</span><span class="syntax--variable syntax--other syntax--member">sample_path</span><span class="syntax--punctuation syntax--section syntax--embedded">}</span>/10500_DN3_H3K4me2/10500_DN3_H3K4me2.bw <span class="syntax--punctuation syntax--section syntax--embedded">${</span><span class="syntax--variable syntax--other syntax--member">sample_path</span><span class="syntax--punctuation syntax--section syntax--embedded">}</span>/ATAC-seq_ETP_T_cells_rep1/ATAC-seq_ETP_T_cells_rep1.bw <span class="syntax--punctuation syntax--section syntax--embedded">${</span><span class="syntax--variable syntax--other syntax--member">sample_path</span><span class="syntax--punctuation syntax--section syntax--embedded">}</span>/ATAC-seq_DN3_T_cells_rep1/ATAC-seq_DN3_T_cells_rep1.bw <span class="syntax--entity syntax--other syntax--attribute-name">-R</span> <span class="syntax--punctuation syntax--section syntax--embedded">${</span><span class="syntax--variable syntax--other syntax--member">sample_path</span><span class="syntax--punctuation syntax--section syntax--embedded">}</span>/merged_peak_files/PU1_peaks.bed <span class="syntax--entity syntax--other syntax--attribute-name">-b</span> 1250 <span class="syntax--entity syntax--other syntax--attribute-name">-a</span> 1250 <span class="syntax--entity syntax--other syntax--attribute-name">--skipZeros</span> <span class="syntax--entity syntax--other syntax--attribute-name">--referencePoint</span> center <span class="syntax--entity syntax--other syntax--attribute-name">-o</span> <span class="syntax--punctuation syntax--section syntax--embedded">${</span><span class="syntax--variable syntax--other syntax--member">sample_path</span><span class="syntax--punctuation syntax--section syntax--embedded">}</span>/</span></span>
<span class=""><span class="syntax--source syntax--shell">chip_atac_matrix.matrix.gz</span></span>
<span class=""></span> 
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">module</span> purge</span></span></pre>
<p>This will create a matrix center in the middle of each peak and extended outwards with 1250 bp in both directions. Read densities will be calculated with a bin size of 10 bp but this can be changed with the <code style="font-family: Menlo, Consolas, &quot;DejaVu Sans Mono&quot;, monospace;">-bs</code> option.</p>
<p>Now are we finally ready to plot the heatmap and while doing so we will also cluster our data into two clusters.</p>
<pre class="editor-colors lang-bash"><span><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#! /bin/bash</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#SBATCH -A lu2018-2-44</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#SBATCH -p lu</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#SBATCH --reservation=lu2018-2-44</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#SBATCH -n 1 # how many processor cores to use</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#SBATCH -N 1 # how many processors to use (always use 1 here unless you know what you are doing)</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#SBATCH -t 00:20:00 # kill the job after ths hh::mm::ss time</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#SBATCH -J 'plotHeatmap' # name of the job</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#SBATCH -o 'plotHeatmap%j.out' # stdout log file</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#SBATCH -e 'plotHeatmap%j.err' # stderr log file</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#SBATCH -p dell # which partition to use</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block"># A script that preprocesses the ChIP samples.</span></span></span>
<span class=""></span> 
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#Load the deepTools module</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">module</span> load GCC/7.3.0-2.30 OpenMPI/3.1.1</span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">module</span> load deepTools/2.5.4-Python-3.6.6</span></span>
<span class=""></span> 
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--variable syntax--other syntax--member">sample_path</span>=<span class="syntax--string">"/projects/fs3/jonun/ChIP-ATAC-workshop-2019-course-material"</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--comment syntax--block">#Plot heatmap</span></span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">plotHeatmap</span> <span class="syntax--entity syntax--other syntax--attribute-name">-m</span> <span class="syntax--punctuation syntax--section syntax--embedded">${</span><span class="syntax--variable syntax--other syntax--member">sample_path</span><span class="syntax--punctuation syntax--section syntax--embedded">}</span>/chip_atac_matrix.matrix.gz <span class="syntax--entity syntax--other syntax--attribute-name">--kmeans</span> 2 <span class="syntax--entity syntax--other syntax--attribute-name">--colorMap</span> jet <span class="syntax--entity syntax--other syntax--attribute-name">-out</span> <span class="syntax--punctuation syntax--section syntax--embedded">${</span><span class="syntax--variable syntax--other syntax--member">sample_path</span><span class="syntax--punctuation syntax--section syntax--embedded">}</span>/workshop_chip_atac_heatmap.png <span class="syntax--entity syntax--other syntax--attribute-name">-T</span> workshop_chip_atac_heatmap</span></span>
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">module</span> purge</span></span></pre>
<p>Behold! Peaks separated between these clusters can furter be targeted for downstream analysis.</p>
<h3>Summary</h3>
<blockquote>
<p><strong>1.</strong> TF and histone peak finding with Homer.</p>
</blockquote>
<blockquote>
<p><strong>2.</strong> Use Homer’s mergePeaks to derive unique peak subsets.</p>
</blockquote>
<blockquote>
<p><strong>3.</strong> Peak annotation with <a href="http://annotatePeaks.pl">annotatePeaks.pl</a>.</p>
</blockquote>
<blockquote>
<p><strong>4.</strong> Motif analysis with Homer and MEME.</p>
</blockquote>
<blockquote>
<p><strong>5.</strong> ATAC-seq peak calling with MACS2.</p>
</blockquote>
<blockquote>
<p><strong>6.</strong> Peak manipulation with bedtools.</p>
</blockquote>
<blockquote>
<p><strong>7.</strong> ChIP/ATAC peak overlap with mergePeaks.</p>
</blockquote>
<blockquote>
<p><strong>8.</strong> Create bigwigs and a basic heatmap with deepTools.</p>
</blockquote>
<p><strong>BONUS:</strong> <strong>FRIP-score calculation with Homer.</strong></p>
<p>We will start by calculating the reads in peaks. We can do this with <code style="font-family: Menlo, Consolas, &quot;DejaVu Sans Mono&quot;, monospace;">annotatePeaks.pl</code> if we provide two extra arguments `-raw’ and ‘-d /path-to-tag-directory’</p>
<pre class="editor-colors lang-bash"><span><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">module</span> load GCC/7.3.0-2.30 homer/4.10</span></span>
<span class=""></span> 
<span class=""><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">annotatePeaks.pl</span> 11307_DN1_PU1_peaks.txt mm10 <span class="syntax--entity syntax--other syntax--attribute-name">-raw</span> <span class="syntax--entity syntax--other syntax--attribute-name">-d</span> ~/NAS2/ChIP-ATAC-workshop-2019-course-material/11307_DN1_PU1/11307_DN1_PU1_tagdir <span class="syntax--keyword syntax--operator">&gt;</span> 11307_DN1_PU1_peaks.counts.txt</span></span></pre>
<p>This will create an annotate peak file with a new column (column 21) containing the read counts in all peaks. We can use awk to summarise this:</p>
<p><code style="font-family: Menlo, Consolas, &quot;DejaVu Sans Mono&quot;, monospace;">awk -F "\t" '{sum += $20} END {print sum}' 11307_DN1_PU1_peaks.counts.txt</code></p>
<blockquote>
<p>Reads in peaks = 1877944</p>
</blockquote>
<p>The total tags/reads can be found in the tagInfo.txt file in the tag directory. Note that the counts will be halfed if the data is paired-end.</p>
<blockquote>
<p>Total tags/reads = 8848637</p>
</blockquote>
<blockquote>
<p>FRIP = 1877944/8848637 = 0.21</p>
</blockquote>
<p>There are many more ways to do this if Homer is not availiable. For instance featureCounts from the subread package can be used if providing a bam and a peak-file. Experiment away!</p>

  </body>
</html>
